{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## basic classifier for Dna2Vec represenation\n",
    "\n",
    "Basic bacterial genome classification using Dna2Vec representation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "import sys\n",
    "sys.path.append(\"..\")\n",
    "from faigen.data import sequence \n",
    "from faigen.data.sequence import regex_filter, count_filter, Dna2VecDataBunch\n",
    "from functools import partial\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn import manifold,neighbors\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage  \n",
    "from matplotlib import pyplot as plt\n",
    "import seaborn as sns; sns.set(color_codes=True)\n",
    "import plotly.plotly as py\n",
    "import plotly.graph_objs as go\n",
    "from fastai import *\n",
    "from fastai.data_block import *\n",
    "from fastai.basic_train import *\n",
    "from fastai.layers import *\n",
    "from fastai.metrics import *\n",
    "from gensim.models import Word2Vec\n",
    "import torch \n",
    "import torch.nn as nn\n",
    "import torch.nn.functional as F"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "if __name__=='__main__':"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "print(\"Loading embedding\")\n",
    "word_vectors = Word2Vec.load_word2vec_format('../faigen/pretrained/embeddings/dna2vec-20190611-1940-k8to8-100d-10c-4870Mbp-sliding-LmP.w2v') "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Data\n",
      "Creating Learner\n",
      "SequentialEx(\n",
      "  (layers): ModuleList(\n",
      "    (0): Linear(in_features=100, out_features=10, bias=True)\n",
      "    (1): ReLU()\n",
      "    (2): Linear(in_features=10, out_features=4, bias=True)\n",
      "  )\n",
      ")\n",
      "Training\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: left;\">\n",
       "      <th>epoch</th>\n",
       "      <th>train_loss</th>\n",
       "      <th>valid_loss</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1.143348</td>\n",
       "      <td>4.225197</td>\n",
       "      <td>0.174679</td>\n",
       "      <td>00:01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>0.997950</td>\n",
       "      <td>3.110410</td>\n",
       "      <td>0.277244</td>\n",
       "      <td>00:01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>0.932375</td>\n",
       "      <td>3.156275</td>\n",
       "      <td>0.280449</td>\n",
       "      <td>00:01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#export\n",
    "print(\"Loading Data\")\n",
    "DB=\"/data/genomes/GenSeq_fastas\"\n",
    "# DB='/home/serge/development/genomes/ncbi-genomes-2019-04-07/bacterial genomes'\n",
    "\n",
    "filters=[partial(regex_filter, rx=\"Bacillus|Staphylococcus|Vibrio|Rhizobium\"),partial(regex_filter, rx=\"plasmid?\\s\", keep=False)]    \n",
    "#        partial(count_filter,num_fastas=(1,1), keep=1)]\n",
    "\n",
    "bunch = Dna2VecDataBunch.from_folder(DB,test=\"test\",\n",
    "             filters=filters,\n",
    "             labeler=lambda x: x.split()[1],\n",
    "             emb=word_vectors,ngram=8,skip=0,\n",
    "             n_cpus=7,agg=partial(np.mean, axis=0))\n",
    "\n",
    "print(\"Creating Learner\")\n",
    "layers=[nn.Linear(bunch.train_dl.x.c,10),nn.ReLU(),\n",
    "        nn.Linear(10,bunch.train_dl.y.c)]\n",
    "bac_classifier = SequentialEx(*layers)\n",
    "print(bac_classifier)\n",
    "learn = Learner(bunch, bac_classifier, metrics=[accuracy])\n",
    "\n",
    "print (\"Training\")\n",
    "learn.fit_one_cycle(3,5e-2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Category 1"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bunch.valid_ds.y[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Dna2VecDataBunch;\n",
       "\n",
       "Train: LabelList (1915 items)\n",
       "x: Dna2VecList\n",
       "[-0.134055 -0.349299 -0.044119  0.068639 ... -0.197828 -0.005512  0.006151 -0.328752],[-0.150918 -0.353403 -0.022866  0.06943  ... -0.197418 -0.00887  -0.01249  -0.324543],[-0.078643 -0.322992 -0.058123  0.041391 ... -0.209444  0.038987 -0.011716 -0.264936],[ 0.016499 -0.299516 -0.075189  0.046569 ... -0.203351 -0.135401  0.028634 -0.259057],[-0.009767 -0.192923 -0.102343  0.078391 ... -0.251641  0.015452  0.024233 -0.29255 ]\n",
       "y: CategoryList\n",
       "0,2,2,2,2\n",
       "Path: /data/genomes/GenSeq_fastas/train;\n",
       "\n",
       "Valid: LabelList (624 items)\n",
       "x: Dna2VecList\n",
       "[-0.134055 -0.349299 -0.044119  0.068639 ... -0.197828 -0.005512  0.006151 -0.328752],[-0.150918 -0.353403 -0.022866  0.06943  ... -0.197418 -0.00887  -0.01249  -0.324543],[-0.167104 -0.27828  -0.016269  0.137708 ... -0.189737  0.02924  -0.064984 -0.335644],[-0.113175 -0.373836  0.001756  0.037688 ... -0.243525 -0.009868 -0.010606 -0.306397],[-0.111694 -0.353767  0.0023    0.053882 ... -0.208598 -0.009581 -0.059168 -0.290926]\n",
       "y: CategoryList\n",
       "1,1,1,1,1\n",
       "Path: /data/genomes/GenSeq_fastas/valid;\n",
       "\n",
       "Test: LabelList (118 items)\n",
       "x: Dna2VecList\n",
       "[-0.025357 -0.3127   -0.018112  0.038752 ... -0.211408  0.032451  0.013458 -0.306945],[-0.040243 -0.218035 -0.051538 -0.028198 ... -0.114473  0.046468 -0.025211 -0.342574],[-0.02691  -0.197986 -0.062984  0.031552 ... -0.202349  0.082913  0.030541 -0.313314],[-0.003488 -0.294841 -0.033978  0.073872 ... -0.264509  0.0134    0.046339 -0.39874 ],[-0.027861 -0.226971 -0.056225  0.044963 ... -0.120433  0.003049  0.046341 -0.314859]\n",
       "y: EmptyLabelList\n",
       ",,,,\n",
       "Path: /data/genomes/GenSeq_fastas/valid"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bunch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "        <style>\n",
       "            /* Turns off some styling */\n",
       "            progress {\n",
       "                /* gets rid of default border in Firefox and Opera. */\n",
       "                border: none;\n",
       "                /* Needs to be in here for Safari polyfill so background images work as expected. */\n",
       "                background-size: auto;\n",
       "            }\n",
       "            .progress-bar-interrupted, .progress-bar-interrupted::-webkit-progress-bar {\n",
       "                background: #F44336;\n",
       "            }\n",
       "        </style>\n",
       "      <progress value='0' class='' max='3', style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      0.00% [0/3 00:00<00:00]\n",
       "    </div>\n",
       "    \n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: left;\">\n",
       "      <th>epoch</th>\n",
       "      <th>train_loss</th>\n",
       "      <th>valid_loss</th>\n",
       "      <th>accuracy</th>\n",
       "      <th>time</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table><p>\n",
       "\n",
       "    <div>\n",
       "        <style>\n",
       "            /* Turns off some styling */\n",
       "            progress {\n",
       "                /* gets rid of default border in Firefox and Opera. */\n",
       "                border: none;\n",
       "                /* Needs to be in here for Safari polyfill so background images work as expected. */\n",
       "                background-size: auto;\n",
       "            }\n",
       "            .progress-bar-interrupted, .progress-bar-interrupted::-webkit-progress-bar {\n",
       "                background: #F44336;\n",
       "            }\n",
       "        </style>\n",
       "      <progress value='0' class='progress-bar-interrupted' max='10', style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      Interrupted\n",
       "    </div>\n",
       "    "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "RuntimeError",
     "evalue": "Expected object of scalar type Long but got scalar type Float for argument #2 'other'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-182-ed8b4cec5453>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mlearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit_one_cycle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m5e-2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/train.py\u001b[0m in \u001b[0;36mfit_one_cycle\u001b[0;34m(learn, cyc_len, max_lr, moms, div_factor, pct_start, final_div, wd, callbacks, tot_epochs, start_epoch)\u001b[0m\n\u001b[1;32m     20\u001b[0m     callbacks.append(OneCycleScheduler(learn, max_lr, moms=moms, div_factor=div_factor, pct_start=pct_start,\n\u001b[1;32m     21\u001b[0m                                        final_div=final_div, tot_epochs=tot_epochs, start_epoch=start_epoch))\n\u001b[0;32m---> 22\u001b[0;31m     \u001b[0mlearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcyc_len\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmax_lr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwd\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mwd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcallbacks\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mlr_find\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlearn\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mLearner\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstart_lr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mFloats\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1e-7\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mend_lr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mFloats\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_it\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstop_div\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mbool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwd\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mfloat\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/basic_train.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, epochs, lr, wd, callbacks)\u001b[0m\n\u001b[1;32m    197\u001b[0m         \u001b[0mcallbacks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mcb\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mcb\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcallback_fns\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mlistify\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdefaults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextra_callback_fns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mlistify\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    198\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mdefaults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextra_callbacks\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mcallbacks\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mdefaults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextra_callbacks\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 199\u001b[0;31m         \u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mepochs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmetrics\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmetrics\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcallbacks\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    200\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    201\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mcreate_opt\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mFloats\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwd\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mFloats\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m->\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/basic_train.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(epochs, learn, callbacks, metrics)\u001b[0m\n\u001b[1;32m    104\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mcb_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mskip_validate\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mlearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mempty_val\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    105\u001b[0m                 val_loss = validate(learn.model, learn.data.valid_dl, loss_func=learn.loss_func,\n\u001b[0;32m--> 106\u001b[0;31m                                        cb_handler=cb_handler, pbar=pbar)\n\u001b[0m\u001b[1;32m    107\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mval_loss\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    108\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mcb_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mon_epoch_end\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mval_loss\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/basic_train.py\u001b[0m in \u001b[0;36mvalidate\u001b[0;34m(model, dl, loss_func, cb_handler, pbar, average, n_batch)\u001b[0m\n\u001b[1;32m     61\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_listy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0myb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0myb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0myb\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m             \u001b[0mnums\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0myb\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 63\u001b[0;31m             \u001b[0;32mif\u001b[0m \u001b[0mcb_handler\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mcb_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mon_batch_end\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mval_losses\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     64\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mn_batch\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnums\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m>=\u001b[0m\u001b[0mn_batch\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m         \u001b[0mnums\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnums\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfloat32\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/callback.py\u001b[0m in \u001b[0;36mon_batch_end\u001b[0;34m(self, loss)\u001b[0m\n\u001b[1;32m    306\u001b[0m         \u001b[0;34m\"Handle end of processing one batch with `loss`.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    307\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'last_loss'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mloss\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 308\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'batch_end'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcall_mets\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'train'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    309\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'train'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    310\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'iteration'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/callback.py\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, cb_name, call_mets, **kwargs)\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[0;34m\"Call through to all of the `CallbakHandler` functions.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    249\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcall_mets\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 250\u001b[0;31m             \u001b[0;32mfor\u001b[0m \u001b[0mmet\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmetrics\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_and_update\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmet\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcb_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    251\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mcb\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcallbacks\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_and_update\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcb_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    252\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/callback.py\u001b[0m in \u001b[0;36m_call_and_update\u001b[0;34m(self, cb, cb_name, **kwargs)\u001b[0m\n\u001b[1;32m    239\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_call_and_update\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcb_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m->\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    240\u001b[0m         \u001b[0;34m\"Call `cb_name` on `cb` and update the inner state.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 241\u001b[0;31m         \u001b[0mnew\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mifnone\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34mf'on_{cb_name}'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    242\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mnew\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mk\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstate_dict\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/callback.py\u001b[0m in \u001b[0;36mon_batch_end\u001b[0;34m(self, last_output, last_target, **kwargs)\u001b[0m\n\u001b[1;32m    342\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_listy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlast_target\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mlast_target\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mlast_target\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    343\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcount\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mlast_target\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 344\u001b[0;31m         \u001b[0mval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlast_output\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0mlast_target\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    345\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mworld\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    346\u001b[0m             \u001b[0mval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mval\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclone\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/fastai/metrics.py\u001b[0m in \u001b[0;36maccuracy\u001b[0;34m(input, targs)\u001b[0m\n\u001b[1;32m     27\u001b[0m     \u001b[0minput\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minput\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0mtargs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mview\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 29\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0mtargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfloat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0maccuracy_thresh\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_pred\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mTensor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_true\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mTensor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mthresh\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mfloat\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0.5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msigmoid\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mbool\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m->\u001b[0m\u001b[0mRank0Tensor\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mRuntimeError\u001b[0m: Expected object of scalar type Long but got scalar type Float for argument #2 'other'"
     ]
    }
   ],
   "source": [
    "learn.recorder.plot_losses()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Category 2"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "learn.predict(10)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "tensor(2)"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../exp/nb_BacteriaClassifier.py\r\n",
      "Converted BacteriaClassifier.ipynb to ../exp/nb_BacteriaClassifier.py\r\n"
     ]
    }
   ],
   "source": [
    "!python notebook2script.py BacteriaClassifier.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LR Finder is complete, type {learner_name}.recorder.plot() to see the graph.\n"
     ]
    }
   ],
   "source": [
    "learn.lr_find()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZYAAAESCAYAAADe2fNYAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvOIA7rQAAIABJREFUeJzt3XlcVPXCP/DPObOwyA4DDEqKmoqKiqFJbqkkpChkKl7LJ3+UpVZ2K8vtPqLd7mLr/V2s2/KU5lP33l9eTW+IXbNcLu4miwLuuMHIjsgizHJ+f6BzRVwGnJkzA5/3K1/pnO+c+cy8dD6c7XsESZIkEBERWYkodwAiImpfWCxERGRVLBYiIrIqFgsREVkVi4WIiKyKxUJERFbFYiEiIqtisRARkVWxWIiIyKpYLEREZFUsFiIisioWCxERWRWLhYiIrEopdwB7qqyshclkm8mc/f09UF5eY5N125Iz5mZm+3HG3MxsPaIowNe3U6uf16GKxWSSbFYsN9bvjJwxNzPbjzPmZmZ5cVcYERFZFYuFiIisisVCRERWxWIhIiKrYrEQEZFVsVjIIpLUfs5YISLb6lCnG7fFpZIa/HX7SdQ1GNCoN0FvMEIQBLiqFXBRK+CqUsBVrYSPtytgkqBSijAam05rNkoSFIIAlVKEUilAIYpo1BtxTW9EQ6MRRpMEUQAEQYAoCEDTfwAAQQAkCZBw/UtdAkwSIEGCJAFGowlGkwTj9VMUlQoRSoUAlUKESZJgMEowXB9zM5NJgt5ggt5ogt5ggkIhwmAwNi2TAMNNy4wmk3k9kgQoFQJcVAqoVQqolCJUChHKG/9XCBDFpvcoCoBRksyfgyRJEAQBwvX3qrieU6UUoRDF6483LZNMEgymptc0mSQoRAFKpXj9/YlQigI8PV3R2KCHQmz63ARRgCg0nXMvXv8slQrBnFOtVEAUm9YvoGmcWqmAWiVCrVKY1yOKwvVlTdmUCv7cRdQWLJZ7EQCVUgHf61+oamXTF/e1RmNTSTQaUVXTiPMlNair10NvNEEhCk1fVqLQ9EVuNEGvN0FCUwG4qhVwUSmgUAiQJAkmE2C6aYtAkiRITS8NQWiqGtH8xXz9y/n6l7hC0bTcYDTBYDDBYDRBuF5mN5YLaCqoG+tRK0W4uyihdBfh5qZCQ4Oh6a0KuKUsmgpDoWgqC4NRQqPeiEaDsalkrxeQwdhUQHq9EUajASZJ+k++65+DJF2/jggSrjWaoDc0fS7G66UFSDBJMJeCUiFCEITr5XZ9/YamojRJEvR6U7PPzBYUYlM5uapv/qWEu6sSbi5KuLso4apWwM2l6c+dXFXw8VDD20MN704uUClZTNQxsVjuoYvGA69OH3jPcRqNJ0pLr95xuSQ1bWmIonDHMXK4V25HdCPzjc/UJDVtGZluKmmj0YRGg+l6EV4vIalpK9BoairEBr0JjQZj05aVJJnXc2PLtNFgQkNj0xbmtUYjrjUYUN9owJXyRtRd06O+wYgGvfGOOb3cVfDzcoWflyu6BHmik1qBAG9XBPi4IcDbFW4u/OdH7RP/ZtvJjV1BZD03PlMRAqCQJ4PJ1LT1Wt9gQE29HldqG1BV04iqmgZUVDegovoadOW1yD1XgYbG5iXk4aaCxscVGh83BPu5Q+vfCVr/pv9za4ecGYuF6D6IogB316bdY/7ergA8bzsuIMADZy9UoPzKNZRW1aPsyjWUVdWjtKoeBbpqHMovMe+uVClF9Ozsjb7dfNG3mx+6Bns2HYMjchIsFiI7EAQBXu5qeLmrEab1arFcbzDickU9dOW1OFtUjbxzFdiw6yw27DoLL3cVBvYMwKAHA9C3mx9cVDJtnhFZiMVC5ABUSgVCAz0QGuiBoeFBAIArtY3IO1eB7NNlOHyiBP/O0UGtEvFQLw2i+wejb1c/hztmRwSwWIgclncnNaL7BSO6XzAMRhNOXKzC4eMlOJRfgn25xfD2UGPkAC1iokLh5a6WOy6RGYuFyAkoFSL6dfNDv25+mBnzILJPl2PvscvYsvc8th28iFGDQhA39AH4ebnKHZWIxULkbFRKBaL6BCKqTyB05bVI338eO44UYseRQowfGorJw8N4HIZkxWIhcmJa/054dmJfJIwIw+aMAmzdfwGH8kvw1GO9MLBngNzxqIPiyfJE7UCAtxuendgXi2ZGQqUU8X//kYNPNh/DtUaD3NGoA2KxELUjvR/wxcrkoXhiZBgOHS/Bqm8yUXm1Qe5Y1MGwWIjaGaVCxKThYXhl6gBcrqzD2+sO42JJjdyxqANhsRC1UwN6BGDJU4MBAH/4+hfknauQORF1FCwWonbsgSBP/Oa/ohDg7YrUDUdxtqha7kjUAbBYiNo5X08XvJ40CF6dVPjT+mzoymvljkTtHIuFqAPw9nDBa0mDIArAB/8viwf0yaZYLEQdRJCvO16dPgi11wz44Nss1DfwVGSyDRYLUQfSNdgTL02JQFFpLdbvPCN3HGqnWCxEHUzfbn54bEgodmYW4vj5SrnjUDvEYiHqgJ4Y1R0aH1es3Xr8rrdXJmoLFgtRB+SiUmD24+EoqarHpn+flTsOtTMsFqIOKryrL0YPCsG2Qxd5fQtZFYuFqAOb9mhP+Hi4YO3W4zCZJLnjUDvBYiHqwNxdlUga2xOXSmtwIK9Y7jjUTrBYiDq4qD6BeCDIA9/9+ywMRpPccagdsFuxFBQUICkpCbGxsUhKSsK5c+duOy49PR2TJk1CfHw8Jk2ahLKyMgBAamoqoqOjkZCQgISEBKxcudJe0YnaNVEQ8OToHii7cg27sorkjkPtgN3uIJmSkoKZM2ciISEBmzdvxvLly7Fu3bpmY44ePYrVq1fjq6++gkajwdWrV6FWq83LExMTsWjRIntFJuow+of5oXeoD77few7DI4LhqubNZant7LLFUl5ejry8PMTHxwMA4uPjkZeXh4qK5tN4r127FsnJydBoNAAAT09PuLi42CMiUYcmCAKefLQHqmsb8ePhS3LHISdnl2LR6XQICgqCQqEAACgUCgQGBkKn0zUbd+bMGVy8eBFPPfUUnnjiCXz88ceQpP+cqbJlyxZMmjQJycnJyMzMtEd0og6jZ2dvRD4YgB8OnEdNvV7uOOTEHGp712g04sSJE1izZg0aGxvx3HPPISQkBImJiZgxYwbmzp0LlUqFPXv2YP78+UhPT4evr6/F6/f397BhekCj8bTp+m3FGXMzs208mxiBl9/bgT25xXj68XAAzpH7VswsL7sUi1arRXFxMYxGIxQKBYxGI0pKSqDVapuNCwkJQVxcHNRqNdRqNcaNG4ecnBwkJiaad48BwPDhw6HVanHq1CkMHTrU4hzl5TU2O1dfo/FEaelVm6zblpwxNzPbjrtCwMAeAdi6twBjB4UgROvtFLlv5iyf9c0cNbMoCm36gdwuu8L8/f0RHh6OtLQ0AEBaWhrCw8Ph5+fXbFx8fDwyMjIgSRL0ej3279+PPn36AACKi/9zjn1+fj4KCwsRFhZmj/hEHcrYhzqjuk6Pw8dL5I5CTspuu8JWrFiBxYsX4+OPP4aXlxdWrVoFAJgzZw4WLFiAiIgITJw4EceOHcOECRMgiiJGjBiBqVOnAgA++OAD5ObmQhRFqFQqvPPOO822YojIOvp280Ownzu2/3IJk8c8KHccckKCdPPR8XaOu8JacsbczGx7P/1yCd/8eBLvvzIKvm4OdSj2npztswYcN7ND7wojIufySP9guKgVSMvgzMfUeiwWImrBzUWJEf21+HdWEaprG+WOQ06GxUJEtzX2oc4wGE3Ylc1pXqh1WCxEdFta/04Y1EuDnZmFMJo4OSVZjsVCRHc0cXgYKq824OjZinsPJrqOxUJEdxQVHgQvdxUycnT3Hkx0HYuFiO5IqRDxSH8tsk+X8SA+WYzFQkR3NXyAFkaThP25l+WOQk6CxUJEd9U5oBO6h3jh3zk6dKDrqek+sFiI6J5GRGhRWFaLc5cd7+pwcjwsFiK6p6HhQVApRR7EJ4uwWIjontxdlYjqrcH+vGI06o1yxyEHx2IhIouMiNCivsGAIydL5Y5CDo7FQkQW6d3VFwHerthzlLvD6O5YLERkEVEQMKxfMPLOV6KqpkHuOOTAWCxEZLHofkGQJOBgXvG9B1OHxWIhIotp/Tuha5An9rFY6C5YLETUKsP6BeH85avQldfKHYUcFIuFiFplaHgQBAD7c7nVQrfHYiGiVvH1dEGfrr7Yn3eZU7zQbbFYiKjVovsFo7TqGs4WVcsdhRwQi4WIWu2h3hqolCJ3h9FtsViIqNXcXJQY2DMAB48Xw2DkbYupORYLEbVJdN8gXK3TI7eAty2m5lgsRNQmET384dVJjV1ZRXJHIQfDYiGiNlEqRIwcoEX2mTJUVF+TOw45EBYLEbXZqIEhgATszuZWC/0Hi4WI2kzj44b+3f2xK7uIB/HJjMVCRPfl0cgQXKlpRPbpcrmjkINgsRDRfRnQwx++ni7YlVUodxRyECwWIrovClHE6IEhOFZQgZKqernjkANgsRDRfRs5MASiIHCrhQCwWIjICnw9XTCwpz/+na1Dg94odxySGYuFiKxi/JBQ1NTrkZGjkzsKyYzFQkRW0SvUBz27eGPrgfM89biDs1uxFBQUICkpCbGxsUhKSsK5c+duOy49PR2TJk1CfHw8Jk2ahLKyMgCA0WjEypUrERMTg8ceewzr16+3V3QisoAgCIiP7oqK6gbOetzBKe31QikpKZg5cyYSEhKwefNmLF++HOvWrWs25ujRo1i9ejW++uoraDQaXL16FWq1GgDw/fff48KFC9i2bRuqqqqQmJiI6OhodOnSxV5vgYjuIaK7P0IDPZC+/zwe6R8MURTkjkQysMsWS3l5OfLy8hAfHw8AiI+PR15eHioqms+KunbtWiQnJ0Oj0QAAPD094eLiAqBpS2batGkQRRF+fn6IiYnBDz/8YI/4RGQhQRAwMborLlfU4cjJUrnjkEzsssWi0+kQFBQEhUIBAFAoFAgMDIROp4Ofn5953JkzZ9ClSxc89dRTqKurw2OPPYZ58+ZBEATodDqEhISYx2q1Wly+fLlVOfz9Pazzhu5Ao/G06fptxRlzM7P9tDZ3nL8H/rnnHP51+CLiRnSHINh/q8UZP2tnzHwndtsVZgmj0YgTJ05gzZo1aGxsxHPPPYeQkBAkJiZaZf3l5TUwmWxzj26NxhOlpVdtsm5bcsbczGw/bc09fkgo1m49jh0HzyOiu78Nkt2ZM37WjppZFIU2/UBul11hWq0WxcXFMBqbzm83Go0oKSmBVqttNi4kJARxcXFQq9Xw8PDAuHHjkJOTY15HUdF/ZlDV6XQIDg62R3wiaqVH+gfDz8sFG3aesdkPc+S47FIs/v7+CA8PR1paGgAgLS0N4eHhzXaDAU3HXjIyMiBJEvR6Pfbv348+ffoAAOLi4rB+/XqYTCZUVFRg+/btiI2NtUd8ImolpULE9DE9caGkhlfjd0B2O914xYoV+PrrrxEbG4uvv/4aK1euBADMmTMHR48eBQBMnDgR/v7+mDBhAhITE9GzZ09MnToVAJCQkIAuXbpg/PjxmD59Ol588UWEhobaKz4RtdKQPoEI7+qLjbvP4mpdo9xxyI4ESZI6zHYqj7G05Iy5mdl+7jd3YVktVnx5ECMGaPFMXB8rJrszZ/ysHTWzQx9jIaKOqXNAJ4x7qAt2ZxWhQFctdxyyExYLEdlUwogweHZS4+ttJ2HqODtIOjQWCxHZlJuLEtPH9ECBrhr7jrXu2jNyTiwWIrK5Yf2CEab1woZdZ9DQyGn12zsWCxHZnCgI+NW4B1FV04itB87LHYdsjMVCRHbRs4s3hoYH4ocDF1BRfU3uOGRDLBYispupo3vAJAEbdp2ROwrZEIuFiOwmwMcN44eEYl9uMc4W8fTj9sqhJqEkovZvYnRXZOQU4Y/fHIG7iwJqlQIuagUeHdQZYwd3lmU2ZLIuFgsR2ZWbixKvTBuIfbmXoTeY0Kg3oaSqDt/8eBI5Z8qRPKEPvD1c5I5J98HiYlmzZg2GDRuG8PBwZGVl4de//jUUCgXee+89REZG2jIjEbUzYVovhGm9zH+WJAk/HynEtztOY/mXB/F/Hg/HoAcDZExI98PiYyxr16413wb4/fffx+zZszF37lz8/ve/t1k4IuoYBEHAuIe6YPnsIfDxcEHqhhzkFlTc+4nkkCwulqtXr8LT0xM1NTU4ceIEZs2ahWnTpqGgoMCW+YioA+kc0AlLZz0EbUAnfP59Lq7UNMgdidrA4mLRarU4cuQI0tPTERUVBYVCgZqaGvPthomIrMFFpcC8hH641mjEZ9/n8UZhTsjiYnnzzTexYMECfPLJJ5g/fz4AYMeOHYiIiLBZOCLqmDprPPDUY72Qf74SafvOyR2HWsnig/ejR49GRkZGs8fi4uIQFxdn9VBERCMGaHH8QiU2ZxSgd6gPej/gK3ckspDFWyynT59GWVkZAKC2thZ//vOf8emnn8JgMNgsHBF1XIIgYFZsb2h83PDNjyfRge5J6PQsLpbXX38d1dVNV8quWrUKhw4dQlZWFpYvX26zcETUsbmqlZgwrCsuldbi+IUqueOQhSzeFVZYWIju3btDkiRs374daWlpcHV1xbhx42yZj4g6uGF9g/CPnWew/fBFhHfl7jBnYPEWi1qtRk1NDXJychAcHAw/Pz+o1Wo0NPB0QCKyHbVKgdGDQpB1qgylVfVyxyELWFws8fHxeOaZZ7Bo0SJMmTIFAJCXl2e+aJKIyFbGRDbNIfbzkUtyRyELWLwrbOnSpcjIyIBSqcSwYcMANB1cW7Jkic3CEREBgJ+XK6L6aLA7W4eEEWFwVXOaQ0fWqmnzR4wYgQceeACZmZkoKipCREQEoqOjbZWNiMgs5qFQ1DcYsO/YZbmj0D1YXPslJSV47bXXkJWVBR8fH1RVVWHQoEF4//33ERQUZMuMRETo0dkL3YI9sf2XSxgd2Rkip9d3WBZvsaxYsQJ9+vTBwYMHkZGRgYMHD6JPnz5ISUmxZT4iIgBNu95jorpAV16H9H3neV2LA7O4WH755RcsWrQI7u7uAAB3d3e8+eabyMzMtFk4IqKbDQ0PQlSfQGzcfRb/k5YHvcEodyS6DYuLxdvbG2fONL9P9dmzZ+Hl5XWHZxARWZdSIWJeQj8kjgzDvtxi/PGbTFRe5SUPjsbiYyzPPfccZs+ejalTpyIkJARFRUXYuHEjXnnlFVvmIyJqRhAETB4ehs4BHviftDz87n8PY9msKPh68q6TjsLiLZbp06fjww8/RGVlJXbs2IHKykq88847uHyZZ2gQkf091FuDxU8NRm29AX/ekIMGPXeLOQpBuo8jYI2NjRg4cCDy8/OtmclmystrbHZvB43GE6WlV22ybltyxtzMbD/OkDvrVBlSN+RgcC8N5j3RH0GBXg6f+VaO+jmLogB/f4/WP+9+X5hnZhCRnAY9GIDpY3vil5Ol+G73WbnjEFpxjOVOBJ5LTkQyGz8kFJcr6rBl33n066FBny48qUhO9yyWffv23XGZXq+3ahgiorYQBAFPPdYL5y5fxdr0PLz97FAoFfe9Q4ba6J7FsmzZsrsu12q1VgtDRNRWSoWIhBFh+PM/cnAgrxjDI/jdJJd7FsvPP/9slRcqKCjA4sWLUVVVBR8fH6xatQrdunVrNiY1NRV//etfERgYCAAYPHiw+cr+uy0jIgKAgT38ERbihS37ziO6XzBEkbvq5WC3KUJTUlIwc+ZMJCQkYPPmzVi+fDnWrVvXYlxiYiIWLVp023XcbRkRkSAImB7TC6vWHcbhEyUYGs55DOVgl52Q5eXlyMvLQ3x8PICme7vk5eWhoqLCHi9PRB1IdEQItP7uSNt7DqbrZ63WNxjw1Q/H8b/bTvBMVjuwS7HodDoEBQVBoVAAABQKBQIDA6HT6VqM3bJlCyZNmoTk5OQW85DdbRkREQAoRAETo7viUmktsk+X4UzhFaxYcxC7soqw40ghDh0vkTtiu3dfF0ha6tixY1i0aBG2bNlifmzChAl499130a9fP/NjpaWl8PHxgUqlwp49e7Bw4UKkp6fD19f3rsuIiG5mNJrwwh9/gt5gRFVNIwK8XfHqrwbji38eQ1nVNXy8aCw83dVyx2y37HKMRavVori4GEajEQqFAkajESUlJS3OKNNoNObfDx8+HFqtFqdOncLQoUPvusxSvPK+JWfMzcz244y5NRpPVFTUYsLDD2DN1uMY1jcIT4/vDXdXJZ5+rBfeWnsYH6/PQvKEcLmjmjnq5yzblfeW8Pf3R3h4ONLS0gAAaWlpCA8Ph5+fX7NxxcXF5t/n5+ejsLAQYWFh91xGRHSrEQO0+OPcaDw/uR/cXZt+hn4gyBNxDz+AjBwd8s7xGK+t2O2ssBUrVmDx4sX4+OOP4eXlhVWrVgEA5syZgwULFiAiIgIffPABcnNzIYoiVCoV3nnnHfOWyt2WERHdShAEBPq4tXh88vBuOHyiBOt+OIG3nh0KtUohQ7r2zS7HWBwFd4W15Iy5mdl+nDG3JZnzz1fi3b9lon+YH+Ym9Ddv0cjFUT9nh94VRkTkSMK7+mL2432Qf74Sv//6F5RW1csdqV1hsRBRhzRqYAheSxqEKzUN+O1Xh3HqUpXckdoNFgsRdVjhXX2x7L+i0MlViff+noWSyjq5I7ULLBYi6tCC/dzx5szBEAUB3+44I3ecdoHFQkQdnq+nCyZGd8WRk6U8DdkKWCxERABih4YiwNsVf/vpFIwmk9xxnBqLhYgIgEqpQNLYnigsrcWurCK54zg1FgsR0XWDe2nQ5wEffLf7LGrqeYfctmKxEBFdJwgCZox7EHUNBmzOKJA7jtNisRAR3eSBIE88Oqgzfj5yCRdLauSO45RYLEREt3hiVHd0clXhG94YrE1YLEREt/BwU+HJ0d1x8tIV7M8rvvcTqBkWCxHRbYwcGIIwrSe+/fk06hsMcsdxKiwWIqLbEAUBT4/vjeraRvxzDw/ktwaLhYjoDsK0Xhg5UIvthy+hqKxW7jhOg8VCRHQXT47uAbVKxIZdnEfMUiwWIqK78HRX4/GHuyLzVBmn1rcQi4WI6B4eiwqFt4ca63ee4enHFmCxEBHdg4tagYQRYTh96QqyTpfJHcfhsViIiCwwcoAWwX7u2LDrLGc/vgcWCxGRBRSiiCdH90BRWS32Hr0sdxyHxmIhIrLQ4F4B6BHihU0ZBdAbuNVyJywWIiILCYKAxJHdUXm1AXuP6eSO47BYLERErdC3my/CtJ5I33+ex1rugMVCRNQKgiAgProbSquu4WB+idxxHBKLhYiolQY+GIDOAZ2wZd95mHhdSwssFiKiVhIFAROju6KorBZZp3hdy61YLEREbTAkPBCBPm5I23uOV+PfgsVCRNQGClHEhOiuOHf5KnILKuSO41BYLEREbRTdLxj+Xq7420+n0Kg3yh3HYbBYiIjaSKUUMfvxPtCV12HTv3kzsBtYLERE96FfmB8eHRSCfx28gNOXrsgdxyGwWIiI7tO0MT3h5+WKL7bkoYG7xFgsRET3y81FieQJfVBcWY/vdp+VO47sWCxERFYQ3s0PYwZ3xo+HLuLc5Wq548jKbsVSUFCApKQkxMbGIikpCefOnWsxJjU1FdHR0UhISEBCQgJWrlxpXmY0GrFy5UrExMTgsccew/r16+0VnYjIIlNH90AnNxX+sfOM3FFkpbTXC6WkpGDmzJlISEjA5s2bsXz5cqxbt67FuMTERCxatKjF499//z0uXLiAbdu2oaqqComJiYiOjkaXLl3sEZ+I6J7cXJSYNLwb/rb9FI4VlKN/mL/ckWRhly2W8vJy5OXlIT4+HgAQHx+PvLw8VFRYflFReno6pk2bBlEU4efnh5iYGPzwww+2ikxE1CaPDuqMAG9X/GPHmQ47j5hdtlh0Oh2CgoKgUCgAAAqFAoGBgdDpdPDz82s2dsuWLcjIyIBGo8HLL7+MyMhI8zpCQkLM47RaLS5fbt1d3Pz9Pe7zndydRuNp0/XbijPmZmb7ccbccmd+Jr4f3v/mF+Rfqsajgy3bqyJ3Zmuy264wS8yYMQNz586FSqXCnj17MH/+fKSnp8PX19cq6y8vr4HJZJufIDQaT5SWXrXJum3JGXMzs/04Y25HyBzexQsPBHrgq7Rc9NJ6QqW8+84hR8h8O6IotOkHcrvsCtNqtSguLobR2HR+t9FoRElJCbRabbNxGo0GKpUKADB8+HBotVqcOnXKvI6ioiLzWJ1Oh+DgYHvEJyJqFVEQMHVMD5RduYadmYVyx7E7uxSLv78/wsPDkZaWBgBIS0tDeHh4i91gxcXF5t/n5+ejsLAQYWFhAIC4uDisX78eJpMJFRUV2L59O2JjY+0Rn4io1fp180Pfbr7YlFGAsiv1csexK7vtCluxYgUWL16Mjz/+GF5eXli1ahUAYM6cOViwYAEiIiLwwQcfIDc3F6IoQqVS4Z133oFGowEAJCQkIDs7G+PHjwcAvPjiiwgNDbVXfCKiVhEEAc/E9UHKlwfxRVo+3vhVJERRkDuWXQhSB7qRAI+xtOSMuZnZfpwxt6NlzsjR4cv0fEwb0wOPP9z1tmMcLfMNDn2MhYiooxoeEYyHemuwcddZXCh2vPKwBRYLEZEN3dgl5uGuwuff50FvaP+TVLJYiIhszMNNhWcnhqOwrBbf7z0ndxybY7EQEdlB/zB/RPcLwg8HLqK0qn2fJcZiISKyk6mP9oQoAt/+fFruKDbFYiEishNfTxdMjO6GX06WIv+c5XMlOhsWCxGRHcUOCUWAtyv++tMpGE0muePYBIuFiMiO1CoFpo/picLSWuzKKrr3E5wQi4WIyM4e6q1Bnwd88N3us7hS2yh3HKtjsRAR2ZkgCHhqfG806E34autxtLcJUFgsREQy6BzQCVNGdUfW6TL8fPii3HGsisVCRCST8UNC0auLNz7bdBTlV67JHcdqWCxERDIRRQHJ8X1hMkn4Mj2/3dzKmMVCRCSjQB83JE/uj/zzlUjfd75dHG9hsRARySxuWFdE9dZg4+6zWLP1uNNPVMliISKSmSAImJvQH/GPdENGjg5/+PqIUx9zYbHEe46xAAAPoklEQVQQETkAURQwZVR3vDwlApcr6rBy7SEU6KrljtUmLBYiIgcS2UuD/34mCq5qBd77e1aLcrla14j/3XYCOWfKZUp4bywWIiIHo/XvhEUzB6OTq7JZuRw7W47lXxzEjiOF+Oi7ozhdeEXmpLfHYiEickD+3q54c2YkOrkq8f7fs/DFljx88G02PNxUeGPGIPh6uuDP/8hBcWWd3FFbYLEQETmoAG83vDkzEu6uSuw5ehkxUV2wfHYUwrv54dVpAwEAH36bjat1jjXfGIuFiMiBBXi74Tf/FYWU2UMwM6YXVEoFACDIzx0LnhyAiuoGpG44Cr3BcabgZ7EQETk4r05qdA32bPF4zy7emDOpL04XXsF3u8/KkOz2WCxERE5sSJ9AjInsjH8dvOAwd6VksRARObnpY3siyM8d/7MlH7XX9HLHYbEQETk7F5UCz0/ui+raRqz74YTs842xWIiI2oFuwV5IHBmGQ8dLsPfYZVmzsFiIiNqJxx/uil6hPlj3rxM4fUm+iydZLERE7YQoCpj/RH/4errg//4jG7ryWnlyyPKqRERkE17uarw2fSBEUcCH32bjSk2D3TOwWIiI2plAX3f8etpAVNc14k/rc3Ct0WDX12exEBG1Q2FaL8xL6I9LpTUoKLLv9PtKu74aERHZzcCeAfjzKyPh5mLfr3pusRARtWP2LhXAjsVSUFCApKQkxMbGIikpCefOnbvj2LNnz2LgwIFYtWqV+bHU1FRER0cjISEBCQkJWLlypR1SExFRa9mtylJSUjBz5kwkJCRg8+bNWL58OdatW9dinNFoREpKCmJiYlosS0xMxKJFi+wRl4iI2sguWyzl5eXIy8tDfHw8ACA+Ph55eXmoqGg5Ydpnn32GRx99FN26dbNHNCIisjK7FItOp0NQUBAUiqb7CCgUCgQGBkKn0zUbd/z4cWRkZGD27Nm3Xc+WLVswadIkJCcnIzMz09axiYioDRzmrDC9Xo///u//xh/+8AdzAd1sxowZmDt3LlQqFfbs2YP58+cjPT0dvr6+Fr+Gv7+HNSO3oNG0vF+CM3DG3MxsP86Ym5nlZZdi0Wq1KC4uhtFohEKhgNFoRElJCbRarXlMaWkpLly4gOeffx4AUF1dDUmSUFNTg9/+9rfQaDTmscOHD4dWq8WpU6cwdOhQi3OUl9fAZLLNrJ8ajSdKS6/aZN225Iy5mdl+nDE3M1uPKApt+oHcLsXi7++P8PBwpKWlISEhAWlpaQgPD4efn595TEhICA4cOGD+c2pqKurq6swH64uLixEUFAQAyM/PR2FhIcLCwlqVQxQFK7wb+dZvK86Ym5ntxxlzM7N1tDWTINlp4v4zZ85g8eLFqK6uhpeXF1atWoXu3btjzpw5WLBgASIiIpqNv7VYFi1ahNzcXIiiCJVKhQULFmD06NH2iE5ERK1gt2IhIqKOgVfeExGRVbFYiIjIqlgsRERkVSwWIiKyKhYLERFZFYuFiIisisVCRERWxWIhIiKrYrHcYtWqVRg7dix69+6NkydPtmkdWVlZmDx5MmJjY5GcnIzy8nLzsqqqKrz22muIjY3FxIkTsXr1aofP3Lt3b0yaNMl8k7UTJ044fOYblixZgt69e6O2tvZ+I9s0s8lkQlJSEiZPnozJkyfj2WefxaVLl+47s61zFxQUYNasWYiLi0N8fDyWLFmCa9euOXRmAHj99dcxYsQIq/zdsEbW27nbzRFbc+NEWUjUzKFDh6SioiJpzJgx0okTJ1r9fJPJJMXExEiHDh2SJEmSPvroI2nx4sXm5S+88IK0Zs0a859LS0sdPnOvXr2kmpqa+855M1tnliRJ+umnn6QlS5ZYLb+tM1dXV5t/v3btWunFF1+878ySZNvcFy9elHJzcyVJkiSj0Si98sor0urVqx06syRJ0t69e6WysjKr/N2436xjxoy57eOzZs2SNm3aJEmSJG3atEmaNWuWRcscAbdYbhEVFdVs1uUbsrOzMWvWLEyZMgVTpkzBzp07b/v8o0ePwsXFBVFRUQCapvv/4YcfAADnzp3DyZMn8cwzz5jHBwQEOHRmW7F15srKSqxevRpLlixxmsyenv+ZNr2mpgaiaJ1/nrbM3aVLF/Tt2xcAIIoiBgwYgKKiIofODADR0dHw9/e/75zWyHo7d7s5YmtunCgXh7kfiyOrrq5GSkoKPvvsMwQGBqKkpARTp05FWloavLy8mo3V6XQICQkx/9nPzw8mkwlVVVU4ffo0goKCsGzZMuTn5yMgIABvvvkmHnzwQYfN7OPjAwCYNWsWjEYjRo0ahZdffhlqtdqhM7/11lt4+eWXm31Z24K1P+c5c+YgLy8Pvr6++OKLL5wmNwBcu3YNGzZswGuvveY0mW2lNVlv5243R5Qk6Y7Lbp4xXk4sFgtkZmbi0qVLmDNnjvkxQRBw/vz5FrMy343RaER2djZef/11REVFYdu2bZg3bx62b9/usJkBYOfOndBqtaipqcEbb7yBjz76CK+++qq1I1st89atW6FSqTBmzBirZ7yVNT9nAPj8889hMpnw6aef4i9/+QtWrFhhxbT/Ye3cBoMBr776KoYNG4Zx48ZZM6qZtTPb0r2yTpkyBUajEQBQUlKChIQEAE33rvrkk09kyWxNLBYLSJKE3r1745tvvmmx7C9/+Yt5E3vJkiXQarXNdgVUVFRAEAT4+PggJCQEWq3WvHk+fvx4vPHGG6ioqLD6TxrWygzAvJnv4eGBadOmYc2aNVbNau3MBw4cwP79+zF27Fjz8vj4eHz++efo2bOnQ2a+mSiKmDp1KsaPH2+zYrFmbqPRiIULF8Lb2xu/+c1vbJLX2plt7W5ZAWDjxo3m348dOxabN29utvxuN0eUJOmeN06UnWxHdxzczQfiqqqqpOHDh0v79u0zL8/OzpZMJlOL5xmNRmncuHG3PWhoMpmk+Ph46eTJk5IkSdLBgwelkSNH3nY9jpK5qqpKqq+vlyRJkvR6vfTmm29Kv/vd76yS11aZb2Xtkw9skbm8vFyqqKgwj123bp2UlJRktcy2ym00GqWFCxdKr732mmQwGKya11aZb2bNvxttzXqng/dPP/10swP0Tz/9tEXLHAHvx3KLt99+G9u2bUNZWRl8fX3h4+ODLVu2ICcnB++++y6uXLkCvV6P0NBQfPLJJ7c9wHrkyBGkpKSgoaEBnTt3xrvvvms+SH/06FGsXLkSjY2NcHNzw7JlyzBgwACHzZyZmYnly5dDEAQYDAZERkZi6dKl6NSpk8NmvlXv3r1x5MgRh8584sQJLFmyBHq9HgDQuXNnLFu2DKGhofeV2da5d+7ciRdeeAG9evUyP2/w4MFISUlx2MwA8NJLLyEnJwfFxcUIDAxEr1692nxM636zjh07Fj///HOL9d7p5oj3WuYIWCxERGRVPN2YiIisisVCRERWxWIhIiKrYrEQEZFVsViIiMiqWCxEdvDcc8/hu+++kzsGkV3wdGNq18aOHYu3334bjzzyiNxRbOLAgQN45pln4ObmBgAIDAzE888/jyeffNKi56empuL8+fN47733bBmTOhhO6UJ0nwwGA5RK+f4pBQYGYvfu3ZAkCbt378a8efMQGRnpUBfMUcfCXWHUYe3YsQMJCQmIiorCjBkzcPz4cfOyzz77DDExMYiMjMSECRPw448/mpdt3LgRM2bMwO9//3sMHToUqamp2LhxI371q19h1apVGDJkCMaOHYtdu3aZnzNr1iysX7/e/Py7jb148SKeeuopREZGYvbs2Vi5ciUWLlx4z/cjCAJGjx4Nb2/vZjdje/vttzF69GgMHjwYU6ZMweHDhwEAu3fvxqeffoqtW7ciMjISkydPBgBcvXoVS5cuxYgRIzBy5Eh8+OGH5gkTiSzBYqEOKTc3F0uXLsVbb72FAwcOICkpCfPnz0djYyMAIDQ0FN988w1++eUXvPTSS3jjjTdQUlJifn5OTg5CQ0Oxd+9ezJs3z/xYWFgY9u/fj+eeew7Lli3DnfY0323swoULMWDAABw4cAAvvfRSiwkK78RkMuGnn35CZWUlunbtan48IiICmzZtwsGDBxEfH49XXnkFDQ0NGDVqFF544QU8/vjjyMzMxD//+U8AwKJFi6BUKrFt2zZs2rQJe/bsMZcikSVYLNQhffvtt0hKSsLAgQOhUCjwxBNPQKVSISsrCwDw+OOPIygoCKIoYsKECejatStycnLMzw8MDMSsWbOgVCrh6uoKAAgJCcH06dPN6ystLUVZWdltX/9OY4uKinD06FEsWLAAarUaUVFRzWZpvp2SkhJERUVhwIABeOmll7B48WLzzbcAICEhAb6+vlAqlUhOTkZjYyMKCgpuu66ysjLs3r0bS5cuhbu7O/z9/TF79mxs2bKlVZ8vdWw8xkIdUlFRETZt2oSvv/7a/JherzdvlWzatAlr1qxBYWEhAKCurg6VlZXmscHBwS3WefMEmDcOptfV1d329e80trKyEt7e3ubHgKYp1HU63R3fy41jLI2NjXjvvfewf/9+zJ4927z8yy+/xPr161FSUgJBEFBTU9PsvdysqKgIBoMBI0aMMD9mMpkca0p2cngsFuqQtFot5s6da96NdbPCwkL85je/wdq1axEZGQmFQmG+EdMNgiDYJJdGo8GVK1dQX19vLpe7lcrN1Go1Fi5ciLi4OGzfvh0xMTE4fPgwPv/8c6xduxYPPvggRFHEkCFDzLvdbn0fwcHBUKvV2L9/v6wnJJBz464wavf0ej0aGhrMvwwGA6ZNm4a///3vyM7OhiRJqKurw86dO1FTU4P6+noIgmC++dqGDRtw6tQpu2Tt3Lkz+vfvj9TUVDQ2NiIzMxM7duyw+PlqtRrJycn46KOPAAC1tbVQKBTw8/ODwWDA6tWrUVNTYx7v7++PwsJCmEwmAE1bP8OHD8cf//hH1NTUwGQy4cKFCzh48KB13yi1aywWaveef/55DBgwwPwrNTUVERER+O1vf4u33noLQ4YMwfjx48139evZsyeSk5MxY8YMPPLIIzh58iQGDx5st7zvvfcesrKy8PDDD+NPf/oTJkyYALVabfHzn3zySRQVFeHnn3/GiBEjMGrUKMTGxmLs2LFwcXFptlsrLi4OAPDwww/jiSeeAAC888470Ov1mDBhAoYMGYIFCxagtLTUum+S2jVeIEnk4H7961+je/fuWLBggdxRiCzCLRYiB5OTk4MLFy7AZDJh9+7d+OmnnxATEyN3LCKL8egckYMpKyvDyy+/jKqqKgQHB2PFihXNTh8mcnTcFUZERFbFXWFERGRVLBYiIrIqFgsREVkVi4WIiKyKxUJERFbFYiEiIqv6/6uUTPqVs6Y9AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "learn.recorder.plot()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "> \u001b[0;32m/home/serge/anaconda3/envs/bio-2/lib/python3.6/site-packages/torch/utils/data/dataloader.py\u001b[0m(608)\u001b[0;36m_process_next_batch\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m    606 \u001b[0;31m                \u001b[0;32mraise\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"KeyError:\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m    607 \u001b[0;31m            \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m--> 608 \u001b[0;31m                \u001b[0;32mraise\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbatch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexc_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m    609 \u001b[0;31m        \u001b[0;32mreturn\u001b[0m \u001b[0mbatch\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0m\u001b[0;32m    610 \u001b[0;31m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0m\n",
      "ipdb> batch\n",
      "<torch.utils.data._utils.ExceptionWrapper object at 0x7ff4d40b8b38>\n"
     ]
    }
   ],
   "source": [
    "%debug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1686"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data.items)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Genome vectors"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Inventory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_fastas = sequence.Dna2VecList.from_folder(DB).items"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "inventory = pd.DataFrame(data=all_fastas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>file</th>\n",
       "      <th>id</th>\n",
       "      <th>len</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NZ_FQXZ01000077.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000077.1</td>\n",
       "      <td>3323</td>\n",
       "      <td>NZ_FQXZ01000077.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NZ_FQXZ01000076.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000076.1</td>\n",
       "      <td>752</td>\n",
       "      <td>NZ_FQXZ01000076.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NZ_FQXZ01000073.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000073.1</td>\n",
       "      <td>735</td>\n",
       "      <td>NZ_FQXZ01000073.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NZ_FQXZ01000071.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000071.1</td>\n",
       "      <td>2512</td>\n",
       "      <td>NZ_FQXZ01000071.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NZ_FQXZ01000070.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000070.1</td>\n",
       "      <td>1089</td>\n",
       "      <td>NZ_FQXZ01000070.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NZ_FQXZ01000068.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000068.1</td>\n",
       "      <td>1662</td>\n",
       "      <td>NZ_FQXZ01000068.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NZ_FQXZ01000067.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000067.1</td>\n",
       "      <td>4452</td>\n",
       "      <td>NZ_FQXZ01000067.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NZ_FQXZ01000066.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000066.1</td>\n",
       "      <td>6487</td>\n",
       "      <td>NZ_FQXZ01000066.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NZ_FQXZ01000062.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000062.1</td>\n",
       "      <td>9215</td>\n",
       "      <td>NZ_FQXZ01000062.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NZ_FQXZ01000061.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000061.1</td>\n",
       "      <td>1952</td>\n",
       "      <td>NZ_FQXZ01000061.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>NZ_FQXZ01000060.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000060.1</td>\n",
       "      <td>11177</td>\n",
       "      <td>NZ_FQXZ01000060.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NZ_FQXZ01000056.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000056.1</td>\n",
       "      <td>5917</td>\n",
       "      <td>NZ_FQXZ01000056.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NZ_FQXZ01000072.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000072.1</td>\n",
       "      <td>1428</td>\n",
       "      <td>NZ_FQXZ01000072.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>NZ_FQXZ01000053.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000053.1</td>\n",
       "      <td>14798</td>\n",
       "      <td>NZ_FQXZ01000053.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NZ_FQXZ01000052.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000052.1</td>\n",
       "      <td>13379</td>\n",
       "      <td>NZ_FQXZ01000052.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NZ_FQXZ01000048.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000048.1</td>\n",
       "      <td>20257</td>\n",
       "      <td>NZ_FQXZ01000048.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NZ_FQXZ01000047.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000047.1</td>\n",
       "      <td>23084</td>\n",
       "      <td>NZ_FQXZ01000047.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>NZ_FQXZ01000063.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000063.1</td>\n",
       "      <td>1371</td>\n",
       "      <td>NZ_FQXZ01000063.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NZ_FQXZ01000020.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000020.1</td>\n",
       "      <td>5620</td>\n",
       "      <td>NZ_FQXZ01000020.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NZ_FQXZ01000017.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000017.1</td>\n",
       "      <td>67412</td>\n",
       "      <td>NZ_FQXZ01000017.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>NZ_FQXZ01000028.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000028.1</td>\n",
       "      <td>4579</td>\n",
       "      <td>NZ_FQXZ01000028.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NZ_FQXZ01000031.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000031.1</td>\n",
       "      <td>27860</td>\n",
       "      <td>NZ_FQXZ01000031.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>NZ_FQXZ01000069.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000069.1</td>\n",
       "      <td>4275</td>\n",
       "      <td>NZ_FQXZ01000069.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>NZ_FQXZ01000050.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000050.1</td>\n",
       "      <td>35273</td>\n",
       "      <td>NZ_FQXZ01000050.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NZ_FQXZ01000014.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000014.1</td>\n",
       "      <td>180465</td>\n",
       "      <td>NZ_FQXZ01000014.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NZ_FQXZ01000054.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000054.1</td>\n",
       "      <td>10520</td>\n",
       "      <td>NZ_FQXZ01000054.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NZ_FQXZ01000001.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000001.1</td>\n",
       "      <td>97699</td>\n",
       "      <td>NZ_FQXZ01000001.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NZ_FQXZ01000005.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000005.1</td>\n",
       "      <td>275127</td>\n",
       "      <td>NZ_FQXZ01000005.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NZ_FQXZ01000035.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000035.1</td>\n",
       "      <td>123878</td>\n",
       "      <td>NZ_FQXZ01000035.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NZ_FQXZ01000074.1 Vibrio aerogenes CECT 7868, ...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/valid/GCF_90013010...</td>\n",
       "      <td>NZ_FQXZ01000074.1</td>\n",
       "      <td>739</td>\n",
       "      <td>NZ_FQXZ01000074.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5269</th>\n",
       "      <td>NZ_FMTM01000027.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000027.1</td>\n",
       "      <td>10431</td>\n",
       "      <td>NZ_FMTM01000027.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5270</th>\n",
       "      <td>NZ_FMTM01000029.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000029.1</td>\n",
       "      <td>6346</td>\n",
       "      <td>NZ_FMTM01000029.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5271</th>\n",
       "      <td>NZ_FMTM01000026.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000026.1</td>\n",
       "      <td>11725</td>\n",
       "      <td>NZ_FMTM01000026.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5272</th>\n",
       "      <td>NZ_FMTM01000010.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000010.1</td>\n",
       "      <td>173930</td>\n",
       "      <td>NZ_FMTM01000010.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5273</th>\n",
       "      <td>NZ_FMTM01000009.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000009.1</td>\n",
       "      <td>235030</td>\n",
       "      <td>NZ_FMTM01000009.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5274</th>\n",
       "      <td>NZ_FMTM01000003.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000003.1</td>\n",
       "      <td>739460</td>\n",
       "      <td>NZ_FMTM01000003.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5275</th>\n",
       "      <td>NZ_FMTM01000006.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000006.1</td>\n",
       "      <td>373336</td>\n",
       "      <td>NZ_FMTM01000006.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5276</th>\n",
       "      <td>NZ_FMTM01000004.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000004.1</td>\n",
       "      <td>484887</td>\n",
       "      <td>NZ_FMTM01000004.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5277</th>\n",
       "      <td>NZ_FMTM01000013.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000013.1</td>\n",
       "      <td>132779</td>\n",
       "      <td>NZ_FMTM01000013.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5278</th>\n",
       "      <td>NZ_FMTM01000005.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000005.1</td>\n",
       "      <td>449193</td>\n",
       "      <td>NZ_FMTM01000005.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5279</th>\n",
       "      <td>NZ_FMTM01000007.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000007.1</td>\n",
       "      <td>358069</td>\n",
       "      <td>NZ_FMTM01000007.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5280</th>\n",
       "      <td>NZ_FMTM01000002.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000002.1</td>\n",
       "      <td>746758</td>\n",
       "      <td>NZ_FMTM01000002.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5281</th>\n",
       "      <td>NZ_FMTM01000008.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000008.1</td>\n",
       "      <td>333892</td>\n",
       "      <td>NZ_FMTM01000008.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5282</th>\n",
       "      <td>NZ_FMTM01000001.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000001.1</td>\n",
       "      <td>1526119</td>\n",
       "      <td>NZ_FMTM01000001.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5283</th>\n",
       "      <td>NZ_FMTM01000024.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000024.1</td>\n",
       "      <td>19925</td>\n",
       "      <td>NZ_FMTM01000024.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5284</th>\n",
       "      <td>NZ_FMTM01000017.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000017.1</td>\n",
       "      <td>96867</td>\n",
       "      <td>NZ_FMTM01000017.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5285</th>\n",
       "      <td>NZ_FMTM01000025.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000025.1</td>\n",
       "      <td>18693</td>\n",
       "      <td>NZ_FMTM01000025.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5286</th>\n",
       "      <td>NZ_FMTM01000023.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000023.1</td>\n",
       "      <td>26879</td>\n",
       "      <td>NZ_FMTM01000023.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5287</th>\n",
       "      <td>NZ_FMTM01000012.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000012.1</td>\n",
       "      <td>151798</td>\n",
       "      <td>NZ_FMTM01000012.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5288</th>\n",
       "      <td>NZ_FMTM01000014.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000014.1</td>\n",
       "      <td>130887</td>\n",
       "      <td>NZ_FMTM01000014.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5289</th>\n",
       "      <td>NZ_FMTM01000016.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000016.1</td>\n",
       "      <td>108628</td>\n",
       "      <td>NZ_FMTM01000016.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5290</th>\n",
       "      <td>NZ_FMTM01000020.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000020.1</td>\n",
       "      <td>71183</td>\n",
       "      <td>NZ_FMTM01000020.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5291</th>\n",
       "      <td>NZ_FMTM01000011.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000011.1</td>\n",
       "      <td>154938</td>\n",
       "      <td>NZ_FMTM01000011.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5292</th>\n",
       "      <td>NZ_FMTM01000015.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000015.1</td>\n",
       "      <td>112264</td>\n",
       "      <td>NZ_FMTM01000015.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5293</th>\n",
       "      <td>NZ_FMTM01000032.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000032.1</td>\n",
       "      <td>1310</td>\n",
       "      <td>NZ_FMTM01000032.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5294</th>\n",
       "      <td>NZ_FMTM01000021.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000021.1</td>\n",
       "      <td>45989</td>\n",
       "      <td>NZ_FMTM01000021.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5295</th>\n",
       "      <td>NZ_FMTM01000030.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000030.1</td>\n",
       "      <td>3775</td>\n",
       "      <td>NZ_FMTM01000030.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5296</th>\n",
       "      <td>NZ_FMTM01000019.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000019.1</td>\n",
       "      <td>74332</td>\n",
       "      <td>NZ_FMTM01000019.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5297</th>\n",
       "      <td>NZ_FMTM01000018.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000018.1</td>\n",
       "      <td>91615</td>\n",
       "      <td>NZ_FMTM01000018.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5298</th>\n",
       "      <td>NZ_FMTM01000022.1 Rhizobium loessense strain C...</td>\n",
       "      <td>/data/genomes/GenSeq_fastas/train/GCF_90009977...</td>\n",
       "      <td>NZ_FMTM01000022.1</td>\n",
       "      <td>26958</td>\n",
       "      <td>NZ_FMTM01000022.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5299 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            description  \\\n",
       "0     NZ_FQXZ01000077.1 Vibrio aerogenes CECT 7868, ...   \n",
       "1     NZ_FQXZ01000076.1 Vibrio aerogenes CECT 7868, ...   \n",
       "2     NZ_FQXZ01000073.1 Vibrio aerogenes CECT 7868, ...   \n",
       "3     NZ_FQXZ01000071.1 Vibrio aerogenes CECT 7868, ...   \n",
       "4     NZ_FQXZ01000070.1 Vibrio aerogenes CECT 7868, ...   \n",
       "5     NZ_FQXZ01000068.1 Vibrio aerogenes CECT 7868, ...   \n",
       "6     NZ_FQXZ01000067.1 Vibrio aerogenes CECT 7868, ...   \n",
       "7     NZ_FQXZ01000066.1 Vibrio aerogenes CECT 7868, ...   \n",
       "8     NZ_FQXZ01000062.1 Vibrio aerogenes CECT 7868, ...   \n",
       "9     NZ_FQXZ01000061.1 Vibrio aerogenes CECT 7868, ...   \n",
       "10    NZ_FQXZ01000060.1 Vibrio aerogenes CECT 7868, ...   \n",
       "11    NZ_FQXZ01000056.1 Vibrio aerogenes CECT 7868, ...   \n",
       "12    NZ_FQXZ01000072.1 Vibrio aerogenes CECT 7868, ...   \n",
       "13    NZ_FQXZ01000053.1 Vibrio aerogenes CECT 7868, ...   \n",
       "14    NZ_FQXZ01000052.1 Vibrio aerogenes CECT 7868, ...   \n",
       "15    NZ_FQXZ01000048.1 Vibrio aerogenes CECT 7868, ...   \n",
       "16    NZ_FQXZ01000047.1 Vibrio aerogenes CECT 7868, ...   \n",
       "17    NZ_FQXZ01000063.1 Vibrio aerogenes CECT 7868, ...   \n",
       "18    NZ_FQXZ01000020.1 Vibrio aerogenes CECT 7868, ...   \n",
       "19    NZ_FQXZ01000017.1 Vibrio aerogenes CECT 7868, ...   \n",
       "20    NZ_FQXZ01000028.1 Vibrio aerogenes CECT 7868, ...   \n",
       "21    NZ_FQXZ01000031.1 Vibrio aerogenes CECT 7868, ...   \n",
       "22    NZ_FQXZ01000069.1 Vibrio aerogenes CECT 7868, ...   \n",
       "23    NZ_FQXZ01000050.1 Vibrio aerogenes CECT 7868, ...   \n",
       "24    NZ_FQXZ01000014.1 Vibrio aerogenes CECT 7868, ...   \n",
       "25    NZ_FQXZ01000054.1 Vibrio aerogenes CECT 7868, ...   \n",
       "26    NZ_FQXZ01000001.1 Vibrio aerogenes CECT 7868, ...   \n",
       "27    NZ_FQXZ01000005.1 Vibrio aerogenes CECT 7868, ...   \n",
       "28    NZ_FQXZ01000035.1 Vibrio aerogenes CECT 7868, ...   \n",
       "29    NZ_FQXZ01000074.1 Vibrio aerogenes CECT 7868, ...   \n",
       "...                                                 ...   \n",
       "5269  NZ_FMTM01000027.1 Rhizobium loessense strain C...   \n",
       "5270  NZ_FMTM01000029.1 Rhizobium loessense strain C...   \n",
       "5271  NZ_FMTM01000026.1 Rhizobium loessense strain C...   \n",
       "5272  NZ_FMTM01000010.1 Rhizobium loessense strain C...   \n",
       "5273  NZ_FMTM01000009.1 Rhizobium loessense strain C...   \n",
       "5274  NZ_FMTM01000003.1 Rhizobium loessense strain C...   \n",
       "5275  NZ_FMTM01000006.1 Rhizobium loessense strain C...   \n",
       "5276  NZ_FMTM01000004.1 Rhizobium loessense strain C...   \n",
       "5277  NZ_FMTM01000013.1 Rhizobium loessense strain C...   \n",
       "5278  NZ_FMTM01000005.1 Rhizobium loessense strain C...   \n",
       "5279  NZ_FMTM01000007.1 Rhizobium loessense strain C...   \n",
       "5280  NZ_FMTM01000002.1 Rhizobium loessense strain C...   \n",
       "5281  NZ_FMTM01000008.1 Rhizobium loessense strain C...   \n",
       "5282  NZ_FMTM01000001.1 Rhizobium loessense strain C...   \n",
       "5283  NZ_FMTM01000024.1 Rhizobium loessense strain C...   \n",
       "5284  NZ_FMTM01000017.1 Rhizobium loessense strain C...   \n",
       "5285  NZ_FMTM01000025.1 Rhizobium loessense strain C...   \n",
       "5286  NZ_FMTM01000023.1 Rhizobium loessense strain C...   \n",
       "5287  NZ_FMTM01000012.1 Rhizobium loessense strain C...   \n",
       "5288  NZ_FMTM01000014.1 Rhizobium loessense strain C...   \n",
       "5289  NZ_FMTM01000016.1 Rhizobium loessense strain C...   \n",
       "5290  NZ_FMTM01000020.1 Rhizobium loessense strain C...   \n",
       "5291  NZ_FMTM01000011.1 Rhizobium loessense strain C...   \n",
       "5292  NZ_FMTM01000015.1 Rhizobium loessense strain C...   \n",
       "5293  NZ_FMTM01000032.1 Rhizobium loessense strain C...   \n",
       "5294  NZ_FMTM01000021.1 Rhizobium loessense strain C...   \n",
       "5295  NZ_FMTM01000030.1 Rhizobium loessense strain C...   \n",
       "5296  NZ_FMTM01000019.1 Rhizobium loessense strain C...   \n",
       "5297  NZ_FMTM01000018.1 Rhizobium loessense strain C...   \n",
       "5298  NZ_FMTM01000022.1 Rhizobium loessense strain C...   \n",
       "\n",
       "                                                   file                 id  \\\n",
       "0     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000077.1   \n",
       "1     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000076.1   \n",
       "2     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000073.1   \n",
       "3     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000071.1   \n",
       "4     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000070.1   \n",
       "5     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000068.1   \n",
       "6     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000067.1   \n",
       "7     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000066.1   \n",
       "8     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000062.1   \n",
       "9     /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000061.1   \n",
       "10    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000060.1   \n",
       "11    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000056.1   \n",
       "12    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000072.1   \n",
       "13    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000053.1   \n",
       "14    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000052.1   \n",
       "15    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000048.1   \n",
       "16    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000047.1   \n",
       "17    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000063.1   \n",
       "18    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000020.1   \n",
       "19    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000017.1   \n",
       "20    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000028.1   \n",
       "21    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000031.1   \n",
       "22    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000069.1   \n",
       "23    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000050.1   \n",
       "24    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000014.1   \n",
       "25    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000054.1   \n",
       "26    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000001.1   \n",
       "27    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000005.1   \n",
       "28    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000035.1   \n",
       "29    /data/genomes/GenSeq_fastas/valid/GCF_90013010...  NZ_FQXZ01000074.1   \n",
       "...                                                 ...                ...   \n",
       "5269  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000027.1   \n",
       "5270  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000029.1   \n",
       "5271  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000026.1   \n",
       "5272  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000010.1   \n",
       "5273  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000009.1   \n",
       "5274  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000003.1   \n",
       "5275  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000006.1   \n",
       "5276  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000004.1   \n",
       "5277  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000013.1   \n",
       "5278  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000005.1   \n",
       "5279  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000007.1   \n",
       "5280  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000002.1   \n",
       "5281  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000008.1   \n",
       "5282  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000001.1   \n",
       "5283  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000024.1   \n",
       "5284  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000017.1   \n",
       "5285  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000025.1   \n",
       "5286  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000023.1   \n",
       "5287  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000012.1   \n",
       "5288  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000014.1   \n",
       "5289  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000016.1   \n",
       "5290  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000020.1   \n",
       "5291  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000011.1   \n",
       "5292  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000015.1   \n",
       "5293  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000032.1   \n",
       "5294  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000021.1   \n",
       "5295  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000030.1   \n",
       "5296  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000019.1   \n",
       "5297  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000018.1   \n",
       "5298  /data/genomes/GenSeq_fastas/train/GCF_90009977...  NZ_FMTM01000022.1   \n",
       "\n",
       "          len               name  \n",
       "0        3323  NZ_FQXZ01000077.1  \n",
       "1         752  NZ_FQXZ01000076.1  \n",
       "2         735  NZ_FQXZ01000073.1  \n",
       "3        2512  NZ_FQXZ01000071.1  \n",
       "4        1089  NZ_FQXZ01000070.1  \n",
       "5        1662  NZ_FQXZ01000068.1  \n",
       "6        4452  NZ_FQXZ01000067.1  \n",
       "7        6487  NZ_FQXZ01000066.1  \n",
       "8        9215  NZ_FQXZ01000062.1  \n",
       "9        1952  NZ_FQXZ01000061.1  \n",
       "10      11177  NZ_FQXZ01000060.1  \n",
       "11       5917  NZ_FQXZ01000056.1  \n",
       "12       1428  NZ_FQXZ01000072.1  \n",
       "13      14798  NZ_FQXZ01000053.1  \n",
       "14      13379  NZ_FQXZ01000052.1  \n",
       "15      20257  NZ_FQXZ01000048.1  \n",
       "16      23084  NZ_FQXZ01000047.1  \n",
       "17       1371  NZ_FQXZ01000063.1  \n",
       "18       5620  NZ_FQXZ01000020.1  \n",
       "19      67412  NZ_FQXZ01000017.1  \n",
       "20       4579  NZ_FQXZ01000028.1  \n",
       "21      27860  NZ_FQXZ01000031.1  \n",
       "22       4275  NZ_FQXZ01000069.1  \n",
       "23      35273  NZ_FQXZ01000050.1  \n",
       "24     180465  NZ_FQXZ01000014.1  \n",
       "25      10520  NZ_FQXZ01000054.1  \n",
       "26      97699  NZ_FQXZ01000001.1  \n",
       "27     275127  NZ_FQXZ01000005.1  \n",
       "28     123878  NZ_FQXZ01000035.1  \n",
       "29        739  NZ_FQXZ01000074.1  \n",
       "...       ...                ...  \n",
       "5269    10431  NZ_FMTM01000027.1  \n",
       "5270     6346  NZ_FMTM01000029.1  \n",
       "5271    11725  NZ_FMTM01000026.1  \n",
       "5272   173930  NZ_FMTM01000010.1  \n",
       "5273   235030  NZ_FMTM01000009.1  \n",
       "5274   739460  NZ_FMTM01000003.1  \n",
       "5275   373336  NZ_FMTM01000006.1  \n",
       "5276   484887  NZ_FMTM01000004.1  \n",
       "5277   132779  NZ_FMTM01000013.1  \n",
       "5278   449193  NZ_FMTM01000005.1  \n",
       "5279   358069  NZ_FMTM01000007.1  \n",
       "5280   746758  NZ_FMTM01000002.1  \n",
       "5281   333892  NZ_FMTM01000008.1  \n",
       "5282  1526119  NZ_FMTM01000001.1  \n",
       "5283    19925  NZ_FMTM01000024.1  \n",
       "5284    96867  NZ_FMTM01000017.1  \n",
       "5285    18693  NZ_FMTM01000025.1  \n",
       "5286    26879  NZ_FMTM01000023.1  \n",
       "5287   151798  NZ_FMTM01000012.1  \n",
       "5288   130887  NZ_FMTM01000014.1  \n",
       "5289   108628  NZ_FMTM01000016.1  \n",
       "5290    71183  NZ_FMTM01000020.1  \n",
       "5291   154938  NZ_FMTM01000011.1  \n",
       "5292   112264  NZ_FMTM01000015.1  \n",
       "5293     1310  NZ_FMTM01000032.1  \n",
       "5294    45989  NZ_FMTM01000021.1  \n",
       "5295     3775  NZ_FMTM01000030.1  \n",
       "5296    74332  NZ_FMTM01000019.1  \n",
       "5297    91615  NZ_FMTM01000018.1  \n",
       "5298    26958  NZ_FMTM01000022.1  \n",
       "\n",
       "[5299 rows x 5 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inventory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'genus'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-99-3220a87f041e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0minventory\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"genus\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0magg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m\"species\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\"count\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"species\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mascending\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36mgroupby\u001b[0;34m(self, by, axis, level, as_index, sort, group_keys, squeeze, observed, **kwargs)\u001b[0m\n\u001b[1;32m   7630\u001b[0m         return groupby(self, by=by, axis=axis, level=level, as_index=as_index,\n\u001b[1;32m   7631\u001b[0m                        \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgroup_keys\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgroup_keys\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msqueeze\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msqueeze\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 7632\u001b[0;31m                        observed=observed, **kwargs)\n\u001b[0m\u001b[1;32m   7633\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   7634\u001b[0m     def asfreq(self, freq, method=None, how=None, normalize=False,\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/pandas/core/groupby/groupby.py\u001b[0m in \u001b[0;36mgroupby\u001b[0;34m(obj, by, **kwds)\u001b[0m\n\u001b[1;32m   2108\u001b[0m         \u001b[0;32mraise\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'invalid type: {}'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2109\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2110\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mklass\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mby\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/pandas/core/groupby/groupby.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, obj, keys, axis, level, grouper, exclusions, selection, as_index, sort, group_keys, squeeze, observed, **kwargs)\u001b[0m\n\u001b[1;32m    358\u001b[0m                                                     \u001b[0msort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    359\u001b[0m                                                     \u001b[0mobserved\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mobserved\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 360\u001b[0;31m                                                     mutated=self.mutated)\n\u001b[0m\u001b[1;32m    361\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    362\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mobj\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bio-2/lib/python3.6/site-packages/pandas/core/groupby/grouper.py\u001b[0m in \u001b[0;36m_get_grouper\u001b[0;34m(obj, key, axis, level, sort, observed, mutated, validate)\u001b[0m\n\u001b[1;32m    576\u001b[0m                 \u001b[0min_axis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlevel\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgpr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    577\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 578\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgpr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    579\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgpr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mGrouper\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mgpr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkey\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    580\u001b[0m             \u001b[0;31m# Add key to exclusions\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'genus'"
     ]
    }
   ],
   "source": [
    "inventory.groupby(\"genus\").agg({\"species\":\"count\"}).sort_values(\"species\",ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'all_fasts' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-6-6d89bdb036cc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mall_fasts\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'all_fasts' is not defined"
     ]
    }
   ],
   "source": [
    "all_fasts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>'Catharanthus</th>\n",
       "      <th>roseus'</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>'Deinococcus</th>\n",
       "      <th>soli'</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>'Nostoc</th>\n",
       "      <th>azollae'</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18,711,729</th>\n",
       "      <th>reads</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acaryochloris</th>\n",
       "      <th>marina</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"10\" valign=\"top\">Acetobacter</th>\n",
       "      <th>aceti</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ascendens</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>orientalis</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oryzifermentans</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pasteurianus</th>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>persici</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pomorum</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>senegalensis</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sp.</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tropicalis</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acetobacterium</th>\n",
       "      <th>woodii</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acetohalobium</th>\n",
       "      <th>arabaticum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acetomicrobium</th>\n",
       "      <th>mobile</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Acholeplasma</th>\n",
       "      <th>axanthum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hippikon</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>laidlawii</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oculi</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">Achromobacter</th>\n",
       "      <th>denitrificans</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>insolitus</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sp.</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spanius</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>xylosoxidans</th>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Acidaminococcus</th>\n",
       "      <th>fermentans</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intestini</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acidiferrobacter</th>\n",
       "      <th>sp.</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Zymobacter</th>\n",
       "      <th>palmae</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Zymomonas</th>\n",
       "      <th>mobilis</th>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Arcobacter]</th>\n",
       "      <th>porcinus</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Bacillus</th>\n",
       "      <th>thuringiensis]</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">[Bacillus]</th>\n",
       "      <th>caldolyticus</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>selenitireducens</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">[Brevibacterium]</th>\n",
       "      <th>flavum</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>frigoritolerans</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">[Clostridium]</th>\n",
       "      <th>bolteae</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cellulolyticum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cellulosi</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>clariflavum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>propionicum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>saccharolyticum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>scindens</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stercorarium</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Enterobacter]</th>\n",
       "      <th>lignolyticus</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">[Eubacterium]</th>\n",
       "      <th>eligens</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hallii</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rectale</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">[Haemophilus]</th>\n",
       "      <th>ducreyi</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>parasuis</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Mycobacterium]</th>\n",
       "      <th>chelonae</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Pasteurella]</th>\n",
       "      <th>aerogenes</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Polyangium]</th>\n",
       "      <th>brachysporum</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Pseudomonas</th>\n",
       "      <th>syringae]</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>[Pseudomonas]</th>\n",
       "      <th>mesoacidophila</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>complete</th>\n",
       "      <th>chromosome</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamma</th>\n",
       "      <th>proteobacterium</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>secondary</th>\n",
       "      <th>endosymbiont</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3186 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   count\n",
       "genus            species                \n",
       "'Catharanthus    roseus'               2\n",
       "'Deinococcus     soli'                 1\n",
       "'Nostoc          azollae'              3\n",
       "18,711,729       reads                 1\n",
       "Acaryochloris    marina               10\n",
       "Acetobacter      aceti                 1\n",
       "                 ascendens             1\n",
       "                 orientalis            2\n",
       "                 oryzifermentans       1\n",
       "                 pasteurianus         91\n",
       "                 persici               2\n",
       "                 pomorum               7\n",
       "                 senegalensis          2\n",
       "                 sp.                   7\n",
       "                 tropicalis            2\n",
       "Acetobacterium   woodii                1\n",
       "Acetohalobium    arabaticum            1\n",
       "Acetomicrobium   mobile                1\n",
       "Acholeplasma     axanthum              1\n",
       "                 hippikon              2\n",
       "                 laidlawii             2\n",
       "                 oculi                 1\n",
       "Achromobacter    denitrificans         3\n",
       "                 insolitus             4\n",
       "                 sp.                   2\n",
       "                 spanius               4\n",
       "                 xylosoxidans         11\n",
       "Acidaminococcus  fermentans            1\n",
       "                 intestini             1\n",
       "Acidiferrobacter sp.                   1\n",
       "...                                  ...\n",
       "Zymobacter       palmae                2\n",
       "Zymomonas        mobilis              49\n",
       "[Arcobacter]     porcinus              1\n",
       "[Bacillus        thuringiensis]        2\n",
       "[Bacillus]       caldolyticus          2\n",
       "                 selenitireducens      1\n",
       "[Brevibacterium] flavum                2\n",
       "                 frigoritolerans       1\n",
       "[Clostridium]    bolteae               2\n",
       "                 cellulolyticum        1\n",
       "                 cellulosi             1\n",
       "                 clariflavum           1\n",
       "                 propionicum           1\n",
       "                 saccharolyticum       1\n",
       "                 scindens              1\n",
       "                 stercorarium          4\n",
       "[Enterobacter]   lignolyticus          1\n",
       "[Eubacterium]    eligens               3\n",
       "                 hallii                1\n",
       "                 rectale               1\n",
       "[Haemophilus]    ducreyi              14\n",
       "                 parasuis              4\n",
       "[Mycobacterium]  chelonae              2\n",
       "[Pasteurella]    aerogenes             1\n",
       "[Polyangium]     brachysporum          1\n",
       "[Pseudomonas     syringae]             3\n",
       "[Pseudomonas]    mesoacidophila        4\n",
       "complete         chromosome            2\n",
       "gamma            proteobacterium       1\n",
       "secondary        endosymbiont          1\n",
       "\n",
       "[3186 rows x 1 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inventory.groupby([\"genus\", \"species\"]).agg({\"species\": \"count\"})\n",
    "inventory.columns=[\"count\"]\n",
    "inventory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NZ_CP013305.1 Vibrio cholerae strain CRC1106 chromosome 1, complete sequence',\n",
       " 'NZ_CP013306.1 Vibrio cholerae strain CRC1106 chromosome 2, complete sequence',\n",
       " 'NC_013791.2 Bacillus pseudofirmus OF4, complete genome',\n",
       " 'NC_013792.1 Bacillus pseudofirmus OF4 plasmid pBpOF4-01, complete sequence',\n",
       " 'NC_013793.1 Bacillus pseudofirmus OF4 plasmid pBpOF4-02, complete sequence',\n",
       " 'NC_000913.3 Escherichia coli str. K-12 substr. MG1655, complete genome',\n",
       " 'NC_007164.1 Corynebacterium jeikeium K411 complete genome',\n",
       " 'NC_003080.1 Corynebacterium jeikeium K411 plasmid pKW4, complete sequence',\n",
       " 'NC_002162.1 Ureaplasma parvum serovar 3 str. ATCC 700970, complete genome',\n",
       " 'NC_004088.1 Yersinia pestis KIM10+, complete genome',\n",
       " 'NC_004838.1 Yersinia pestis KIM10+ plasmid pMT-1, complete sequence',\n",
       " 'NC_002620.2 Chlamydia muridarum Nigg, complete genome',\n",
       " 'NC_002182.1 Chlamydia muridarum Nigg plasmid pMoPn, complete sequence',\n",
       " 'NC_002488.3 Xylella fastidiosa 9a5c, complete genome',\n",
       " 'NC_002489.3 Xylella fastidiosa 9a5c plasmid pXF1.3, complete sequence',\n",
       " 'NC_002490.1 Xylella fastidiosa 9a5c plasmid pXF51, complete sequence',\n",
       " 'NC_002505.1 Vibrio cholerae O1 biovar El Tor str. N16961 chromosome I, complete sequence',\n",
       " 'NC_002506.1 Vibrio cholerae O1 biovar El Tor str. N16961 chromosome II, complete sequence',\n",
       " 'NC_002516.2 Pseudomonas aeruginosa PAO1, complete genome',\n",
       " 'NC_002737.2 Streptococcus pyogenes M1 GAS, complete genome',\n",
       " 'NC_002663.1 Pasteurella multocida subsp. multocida str. Pm70, complete genome',\n",
       " 'NC_002946.2 Neisseria gonorrhoeae FA 1090 chromosome, complete genome',\n",
       " 'NC_002662.1 Lactococcus lactis subsp. lactis Il1403 chromosome, complete genome',\n",
       " 'NC_003028.3 Streptococcus pneumoniae TIGR4, complete genome',\n",
       " 'NC_002696.2 Caulobacter crescentus CB15 chromosome, complete genome',\n",
       " 'NC_004337.2 Shigella flexneri 2a str. 301 chromosome, complete genome',\n",
       " 'NC_004851.1 Shigella flexneri 2a str. 301 plasmid pCP301, complete sequence',\n",
       " 'NC_013174.1 Jonesia denitrificans DSM 20603, complete genome',\n",
       " 'NC_013166.1 Kangiella koreensis DSM 16069, complete genome',\n",
       " 'NC_013171.1 Anaerococcus prevotii DSM 20548, complete genome',\n",
       " 'NC_013164.1 Anaerococcus prevotii DSM 20548 plasmid pAPRE01, complete sequence',\n",
       " 'NC_013222.1 Robiginitalea biformata HTCC2501, complete genome',\n",
       " 'NC_017285.1 Chlamydophila pneumoniae LPCoLN, complete genome',\n",
       " 'NC_017286.1 Chlamydophila pneumoniae LPCoLN plasmid, complete sequence',\n",
       " 'NC_013194.1 Candidatus Accumulibacter phosphatis clade IIA str. UW-1, complete genome',\n",
       " 'NC_013193.1 Candidatus Accumulibacter phosphatis clade IIA str. UW-1 plasmid pAph01, complete sequence',\n",
       " 'NC_013190.1 Candidatus Accumulibacter phosphatis clade IIA str. UW-1 plasmid pAph02, complete sequence',\n",
       " 'NC_013191.1 Candidatus Accumulibacter phosphatis clade IIA str. UW-1 plasmid pAph03, complete sequence',\n",
       " 'NC_013216.1 Desulfotomaculum acetoxidans DSM 771, complete genome',\n",
       " 'NC_013203.1 Atopobium parvulum DSM 20469, complete genome',\n",
       " 'NC_013355.1 Zymomonas mobilis subsp. mobilis NCIMB 11163, complete genome',\n",
       " 'NC_013356.1 Zymomonas mobilis subsp. mobilis NCIMB 11163 plasmid pZA1001, complete sequence',\n",
       " 'NC_013357.1 Zymomonas mobilis subsp. mobilis NCIMB 11163 plasmid pZA1002, complete sequence',\n",
       " 'NC_013358.1 Zymomonas mobilis subsp. mobilis NCIMB 11163 plasmid pZA1003, complete sequence',\n",
       " 'NC_013204.1 Eggerthella lenta DSM 2243, complete genome',\n",
       " 'NC_013205.1 Alicyclobacillus acidocaldarius subsp. acidocaldarius DSM 446, complete genome',\n",
       " 'NC_013206.1 Alicyclobacillus acidocaldarius subsp. acidocaldarius DSM 446 plasmid pAACI01, complete sequence',\n",
       " 'NC_013207.1 Alicyclobacillus acidocaldarius subsp. acidocaldarius DSM 446 plasmid pAACI02, complete sequence',\n",
       " 'NC_013208.1 Alicyclobacillus acidocaldarius subsp. acidocaldarius DSM 446 plasmid pAACI03, complete sequence',\n",
       " 'NC_013223.1 Desulfohalobium retbaense DSM 5692, complete genome',\n",
       " 'NC_013224.1 Desulfohalobium retbaense DSM 5692 plasmid pDRET01, complete sequence',\n",
       " 'NC_009883.1 Rickettsia bellii OSU 85-389, complete genome',\n",
       " 'NC_009953.1 Salinispora arenicola CNS-205, complete genome',\n",
       " 'NC_009901.1 Shewanella pealeana ATCC 700345, complete genome',\n",
       " 'NC_009922.1 Alkaliphilus oremlandii OhILAs, complete genome',\n",
       " 'NC_010612.1 Mycobacterium marinum M, complete genome',\n",
       " 'NC_010604.1 Mycobacterium marinum M plasmid pMM23, complete sequence',\n",
       " 'NC_012125.1 Salmonella enterica subsp. enterica serovar Paratyphi C strain RKS4594, complete genome',\n",
       " 'NC_012124.1 Salmonella enterica subsp. enterica serovar Paratyphi C strain RKS4594 plasmid pSPCV, complete sequence',\n",
       " 'NC_009943.1 Desulfococcus oleovorans Hxd3, complete genome',\n",
       " 'NC_010424.1 Candidatus Desulforudis audaxviator MP104C, complete genome',\n",
       " 'NC_010611.1 Acinetobacter baumannii ACICU, complete genome',\n",
       " 'NC_010605.1 Acinetobacter baumannii ACICU plasmid pACICU1, complete sequence',\n",
       " 'NC_010606.1 Acinetobacter baumannii ACICU plasmid pACICU2, complete sequence',\n",
       " 'NC_010084.1 Burkholderia multivorans ATCC 17616 chromosome 1, complete sequence',\n",
       " 'NC_010086.1 Burkholderia multivorans ATCC 17616 chromosome 2, complete sequence',\n",
       " 'NC_010087.1 Burkholderia multivorans ATCC 17616 chromosome 3, complete sequence',\n",
       " 'NC_010070.1 Burkholderia multivorans ATCC 17616 plasmid pBMUL01, complete sequence',\n",
       " 'NC_010103.1 Brucella canis ATCC 23365 chromosome I, complete sequence',\n",
       " 'NC_010104.1 Brucella canis ATCC 23365 chromosome II, complete sequence',\n",
       " 'NC_012587.1 Sinorhizobium fredii NGR234 chromosome, complete genome',\n",
       " 'NC_000914.2 Sinorhizobium fredii NGR234 plasmid pNGR234a, complete sequence',\n",
       " 'NC_012586.1 Sinorhizobium fredii NGR234 plasmid pNGR234b, complete sequence',\n",
       " 'NC_010003.1 Petrotoga mobilis SJ95, complete genome',\n",
       " 'NC_010067.1 Salmonella enterica subsp. arizonae serovar 62:z4,z23:--, complete genome',\n",
       " 'NC_010002.1 Delftia acidovorans SPH-1, complete genome',\n",
       " 'NC_010001.1 Lachnoclostridium phytofermentans ISDg, complete genome',\n",
       " 'NC_010102.1 Salmonella enterica subsp. enterica serovar Paratyphi B str. SPB7, complete genome',\n",
       " 'NC_010742.1 Brucella abortus S19 chromosome 1, complete sequence',\n",
       " 'NC_010740.1 Brucella abortus S19 chromosome 2, complete sequence',\n",
       " 'NC_010117.1 Coxiella burnetii RSA 331, complete genome',\n",
       " 'NC_010115.1 Coxiella burnetii RSA 331 plasmid QpH1, complete sequence',\n",
       " 'NC_009997.1 Shewanella baltica OS195, complete genome',\n",
       " 'NC_009998.1 Shewanella baltica OS195 plasmid pS19501, complete sequence',\n",
       " 'NC_009999.1 Shewanella baltica OS195 plasmid pS19502, complete sequence',\n",
       " 'NC_010000.1 Shewanella baltica OS195 plasmid pS19503, complete sequence',\n",
       " 'NC_010163.1 Acholeplasma laidlawii PG-8A, complete genome',\n",
       " 'NC_003197.2 Salmonella enterica subsp. enterica serovar Typhimurium str. LT2, complete genome',\n",
       " 'NC_003277.2 Salmonella enterica subsp. enterica serovar Typhimurium str. LT2 plasmid pSLT, complete sequence',\n",
       " 'NC_004631.1 Salmonella enterica subsp. enterica serovar Typhi Ty2, complete genome',\n",
       " 'NC_002944.2 Mycobacterium avium subsp. paratuberculosis str. k10, complete genome',\n",
       " 'NC_007332.1 Mycoplasma hyopneumoniae 7448, complete genome',\n",
       " 'NC_005957.1 [Bacillus thuringiensis] serovar konkukian str. 97-27 chromosome, complete genome',\n",
       " 'NC_006578.1 [Bacillus thuringiensis] serovar konkukian str. 97-27 plasmid pBT9727, complete sequence',\n",
       " 'NC_010816.1 Bifidobacterium longum DJO10A, complete genome',\n",
       " 'NC_004252.1 Bifidobacterium longum DJO10A plasmid pDOJH10L, complete sequence',\n",
       " 'NC_004253.1 Bifidobacterium longum DJO10A plasmid pDOJH10S, complete sequence',\n",
       " 'NC_008800.1 Yersinia enterocolitica subsp. enterocolitica 8081 chromosome, complete genome',\n",
       " 'NC_008791.1 Yersinia enterocolitica subsp. enterocolitica 8081 plasmid pYVe8081, complete sequence',\n",
       " 'NC_003272.1 Nostoc sp. PCC 7120 DNA, complete genome',\n",
       " 'NC_003276.1 Nostoc sp. PCC 7120 plasmid pCC7120alpha DNA, complete genome',\n",
       " 'NC_003240.1 Nostoc sp. PCC 7120 plasmid pCC7120beta DNA, complete sequence',\n",
       " 'NC_003273.1 Nostoc sp. PCC 7120 plasmid pCC7120delta DNA, complete sequence',\n",
       " 'NC_003270.1 Nostoc sp. PCC 7120 plasmid pCC7120epsilon DNA, complete sequence',\n",
       " 'NC_003267.1 Nostoc sp. PCC 7120 plasmid pCC7120gamma DNA, complete sequence',\n",
       " 'NC_003241.1 Nostoc sp. PCC 7120 plasmid pCC7120zeta DNA, complete sequence',\n",
       " 'NC_009342.1 Corynebacterium glutamicum R DNA, complete genome',\n",
       " 'NC_009343.1 Corynebacterium glutamicum R plasmid pCGR1 DNA, complete sequence',\n",
       " 'NC_010729.1 Porphyromonas gingivalis ATCC 33277 DNA, complete genome',\n",
       " 'NC_013799.1 Hydrogenobacter thermophilus TK-6 DNA, complete genome',\n",
       " 'NC_003450.3 Corynebacterium glutamicum ATCC 13032 chromosome, complete genome',\n",
       " 'NC_017382.1 Helicobacter pylori 51, complete genome',\n",
       " 'NC_008322.1 Shewanella sp. MR-7, complete genome',\n",
       " 'NC_008320.1 Shewanella sp. MR-7, complete sequence',\n",
       " 'NC_008321.1 Shewanella sp. MR-4, complete genome',\n",
       " 'NC_008345.1 Shewanella frigidimarina NCIMB 400, complete genome',\n",
       " 'NC_008346.1 Syntrophomonas wolfei subsp. wolfei str. Goettingen G311, complete genome',\n",
       " 'NC_008347.1 Maricaulis maris MCS10, complete genome',\n",
       " 'NC_008344.1 Nitrosomonas eutropha C91, complete genome',\n",
       " 'NC_008341.1 Nitrosomonas eutropha C91 Plasmid1, complete sequence',\n",
       " 'NC_008342.1 Nitrosomonas eutropha C91 Plasmid2, complete sequence',\n",
       " 'NC_008340.1 Alkalilimnicola ehrlichii MLHE-1, complete genome',\n",
       " 'NC_008570.1 Aeromonas hydrophila subsp. hydrophila ATCC 7966 chromosome, complete genome',\n",
       " 'NC_008435.1 Rhodopseudomonas palustris BisA53, complete genome',\n",
       " 'NC_008563.1 Escherichia coli APEC O1, complete genome',\n",
       " 'NC_009837.1 Escherichia coli APEC O1 plasmid pAPEC-O1-ColBM, complete sequence',\n",
       " 'NC_009838.1 Escherichia coli APEC O1 plasmid pAPEC-O1-R, complete sequence',\n",
       " 'NC_008576.1 Magnetococcus marinus MC-1, complete genome',\n",
       " 'NC_011566.1 Shewanella piezotolerans WP3, complete genome',\n",
       " 'NC_008536.1 Candidatus Solibacter usitatus Ellin6076, complete genome',\n",
       " 'NC_008711.1 Paenarthrobacter aurescens TC1, complete genome',\n",
       " 'NC_008712.1 Paenarthrobacter aurescens TC1 plasmid TC1, complete sequence',\n",
       " 'NC_008713.1 Paenarthrobacter aurescens TC1 plasmid TC2, complete sequence',\n",
       " 'NC_008554.1 Syntrophobacter fumaroxidans MPOB, complete genome',\n",
       " 'NC_008595.1 Mycobacterium avium 104, complete genome',\n",
       " 'NC_008596.1 Mycobacterium smegmatis str. MC2 155 chromosome, complete genome',\n",
       " 'NC_008578.1 Acidothermus cellulolyticus 11B, complete genome',\n",
       " 'NC_008609.1 Pelobacter propionicus DSM 2379, complete genome',\n",
       " 'NC_008607.1 Pelobacter propionicus DSM 2379 plasmid pPRO1, complete sequence',\n",
       " 'NC_008608.1 Pelobacter propionicus DSM 2379 plasmid pPRO2, complete sequence',\n",
       " 'NC_008600.1 Bacillus thuringiensis str. Al Hakam, complete genome',\n",
       " 'NC_008598.1 Bacillus thuringiensis str. Al Hakam, plasmid pALH1, complete sequence',\n",
       " 'NC_013928.1 Streptococcus mutans NN2025 DNA, complete genome',\n",
       " 'NC_014215.1 Propionibacterium freudenreichii subsp. shermanii CIRM-BIA1, complete genome',\n",
       " 'NC_013861.1 Legionella longbeachae NSW150, complete genome',\n",
       " 'NC_014544.1 Legionella longbeachae NSW150 plasmid pLLO',\n",
       " 'NC_012925.1 Streptococcus suis P1/7 complete genome',\n",
       " 'NC_013949.1 Helicobacter mustelae 12198 complete genome',\n",
       " 'NC_003062.2 Agrobacterium fabrum str. C58 chromosome circular, complete sequence',\n",
       " 'NC_003063.2 Agrobacterium fabrum str. C58 chromosome linear, complete sequence',\n",
       " 'NC_003064.2 Agrobacterium fabrum str. C58 plasmid At, complete sequence',\n",
       " 'NC_003065.3 Agrobacterium fabrum str. C58 plasmid Ti, complete sequence',\n",
       " 'NC_007761.1 Rhizobium etli CFN 42, complete genome',\n",
       " 'NC_007762.1 Rhizobium etli CFN 42 plasmid p42a, complete sequence',\n",
       " 'NC_007763.1 Rhizobium etli CFN 42 plasmid p42b, complete sequence',\n",
       " 'NC_007764.1 Rhizobium etli CFN 42 plasmid p42c, complete sequence',\n",
       " 'NC_007765.1 Rhizobium etli CFN 42 plasmid p42e, complete sequence',\n",
       " 'NC_007766.1 Rhizobium etli CFN 42 plasmid p42f, complete sequence',\n",
       " 'NC_004041.2 Rhizobium etli CFN 42 plasmid symbiotic plasmid p42d, complete sequence',\n",
       " 'NC_014148.1 Planctopirus limnophila DSM 3776, complete genome',\n",
       " 'NC_014149.1 Planctopirus limnophila DSM 3776 plasmid pPLIM01, complete sequence',\n",
       " 'NC_014212.1 Meiothermus silvanus DSM 9946, complete genome',\n",
       " 'NC_014213.1 Meiothermus silvanus DSM 9946 plasmid pMESIL01, complete sequence',\n",
       " 'NC_014214.1 Meiothermus silvanus DSM 9946 plasmid pMESIL02, complete sequence',\n",
       " 'NC_014171.1 Bacillus thuringiensis BMB171, complete genome',\n",
       " 'NC_014172.1 Bacillus thuringiensis BMB171 plasmid pBMB171, complete sequence',\n",
       " 'NC_014216.1 Desulfurivibrio alkaliphilus AHT2, complete genome',\n",
       " 'NC_014158.1 Tsukamurella paurometabola DSM 20162, complete genome',\n",
       " 'NC_014159.1 Tsukamurella paurometabola DSM 20162 plasmid pTpau01, complete sequence',\n",
       " 'NC_014166.1 Arcobacter nitrofigilis DSM 7299, complete genome',\n",
       " 'NC_014147.1 Moraxella catarrhalis BBH18, complete genome',\n",
       " 'NC_014100.1 Caulobacter segnis ATCC 21756, complete genome',\n",
       " 'NC_014169.1 Bifidobacterium longum subsp. longum JDM301, complete genome',\n",
       " 'NC_017297.1 Clostridium botulinum F str. 230613, complete genome',\n",
       " 'NC_017298.1 Clostridium botulinum F str. 230613 plasmid pCBF, complete sequence',\n",
       " 'NC_014218.1 Arcanobacterium haemolyticum DSM 20595, complete genome',\n",
       " 'NC_016582.1 Streptomyces bingchenggensis BCW-1, complete genome',\n",
       " 'NC_014220.1 Syntrophothermus lipocalidus DSM 12680, complete genome',\n",
       " 'NC_017633.1 Escherichia coli ETEC H10407, complete genome',\n",
       " 'NC_017721.1 Escherichia coli ETEC H10407 p52 plasmid',\n",
       " 'NC_017723.1 Escherichia coli ETEC H10407 p58 plasmid',\n",
       " 'NC_017722.1 Escherichia coli ETEC H10407 p666 plasmid',\n",
       " 'NC_017724.1 Escherichia coli ETEC H10407 p948 plasmid',\n",
       " 'NC_017441.1 Chlamydia trachomatis Sweden2, complete genome',\n",
       " 'NC_017465.1 Lactobacillus fermentum CECT 5716, complete genome',\n",
       " 'NC_018588.1 Listeria monocytogenes strain SLCC2372, serotype 1/2c',\n",
       " 'NC_018889.1 Listeria monocytogenes serotype 1-2c str. SLCC2372 plasmid pLM1-2cUG1 complete sequence',\n",
       " 'NC_016776.1 Bacteroides fragilis 638R genome',\n",
       " 'NC_016810.1 Salmonella enterica subsp. enterica serovar Typhimurium SL1344 complete genome',\n",
       " 'NC_017718.1 Salmonella enterica subsp. enterica serovar Typhimurium str. SL1344 plasmid pCol1B9_SL1344 complete sequence',\n",
       " 'NC_017719.1 Salmonella enterica subsp. enterica serovar Typhimurium str. SL1344 plasmid pRSF1010_SL1344 complete sequence',\n",
       " 'NC_017720.1 Salmonella enterica subsp. enterica serovar Typhimurium str. SL1344 plasmid pSLT_SL1344 complete sequence',\n",
       " 'NC_016809.1 Haemophilus influenzae 10810 genome',\n",
       " 'NC_015964.1 Haemophilus parainfluenzae T3T1 complete genome',\n",
       " 'NC_016620.1 Bacteriovorax marinus SJ genome',\n",
       " 'NC_019100.1 Bacteriovorax marinus SJ plasmid pBMS1 complete sequence',\n",
       " 'NC_017593.1 Streptococcus pneumoniae INV200 genome',\n",
       " 'NC_017592.1 Streptococcus pneumoniae OXC141 complete genome',\n",
       " 'NC_017591.1 Streptococcus pneumoniae INV104 genome',\n",
       " 'NC_021006.1 Streptococcus pneumoniae SPN034156 draft genome',\n",
       " 'NC_021028.1 Streptococcus pneumoniae SPN034183 draft genome',\n",
       " 'NC_021026.1 Streptococcus pneumoniae SPN994038 draft genome',\n",
       " 'NC_021005.1 Streptococcus pneumoniae SPN994039 draft genome',\n",
       " 'NC_021003.1 Streptococcus pneumoniae SPN032672 draft genome',\n",
       " 'NC_021004.1 Streptococcus pneumoniae SPN033038 draft genome',\n",
       " 'NC_017519.1 Mycoplasma hyorhinis MCLD, complete genome',\n",
       " 'NC_015428.1 Lactobacillus buchneri NRRL B-30929, complete genome',\n",
       " 'NC_015420.1 Lactobacillus buchneri NRRL B-30929 plasmid pLBUC01, complete sequence',\n",
       " 'NC_015429.1 Lactobacillus buchneri NRRL B-30929 plasmid pLBUC02, complete sequence',\n",
       " 'NC_015421.1 Lactobacillus buchneri NRRL B-30929 plasmid pLBUC03, complete sequence',\n",
       " 'NC_023030.2 Mycoplasma gallisepticum S6, complete genome',\n",
       " 'NC_015638.1 Lacinutrix sp. 5H-3-7-4, complete genome',\n",
       " 'NC_015497.1 Glaciecola sp. 4H-3-7+YE-5, complete genome',\n",
       " 'NC_015498.1 Glaciecola sp. 4H-3-7+YE-5 plasmid pGLAAG01, complete sequence',\n",
       " 'NC_015496.1 Dokdonia sp. 4H-3-7-5, complete genome',\n",
       " 'NC_015501.1 Porphyromonas asaccharolytica DSM 20707, complete genome',\n",
       " 'NC_018080.1 Pseudomonas aeruginosa DK2, complete genome',\n",
       " 'NC_018081.1 Enterococcus hirae ATCC 9790, complete genome',\n",
       " 'NC_015845.1 Enterococcus hirae ATCC 9790 plasmid pTG9790, complete sequence',\n",
       " 'NZ_CP010978.1 Pelosinus fermentans JBW45, complete genome',\n",
       " 'NC_018089.1 Streptococcus mutans GS-5, complete genome',\n",
       " 'NZ_CP015923.1 Salmonella enterica subsp. enterica serovar Newport str. Levine 1, complete genome',\n",
       " 'NZ_CP015924.1 Salmonella enterica subsp. enterica serovar Newport str. Levine 15, complete genome',\n",
       " 'NZ_CP015876.1 Pseudomonas putida SJTE-1, complete genome',\n",
       " 'NZ_CP015877.1 Pseudomonas aeruginosa SJTD-1, complete genome',\n",
       " 'NZ_CP010283.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 21550, complete genome',\n",
       " 'NZ_CP009564.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 21550 plasmid pCVM21550, complete sequence',\n",
       " 'NZ_CP010281.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22513, complete genome',\n",
       " 'NZ_CP009562.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22513 plasmid pCVM22513, complete sequence',\n",
       " 'NZ_CP010282.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 21538, complete genome',\n",
       " 'NZ_CP009563.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 21538 plasmid pCVM21538, complete sequence',\n",
       " 'NZ_CP010279.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22425, complete genome',\n",
       " 'NZ_CP009560.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22425 plasmid pCVM22425, complete sequence',\n",
       " 'NZ_CP009561.1 Salmonella enterica subsp. enterica serovar Newport str. CVM N18486, complete genome',\n",
       " 'NZ_CP010284.1 Salmonella enterica subsp. enterica serovar Newport str. CVM N1543, complete genome',\n",
       " 'NZ_CP009570.1 Salmonella enterica subsp. enterica serovar Newport str. CVM N1543 plasmid pCVMN1543, complete sequence',\n",
       " 'NZ_CP010280.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22462, complete genome',\n",
       " 'NZ_CP009567.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22462 plasmid pCFSAN000934_02, complete sequence',\n",
       " 'NZ_CP009568.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22462 plasmid pCFSAN000934_03, complete sequence',\n",
       " 'NZ_CP009566.1 Salmonella enterica subsp. enterica serovar Newport str. CVM 22462 plasmid pCVM22462, complete sequence',\n",
       " 'NC_018108.1 Taylorella equigenitalis ATCC 35865, complete genome',\n",
       " 'NC_018106.1 Klebsiella michiganensis E718, complete sequence',\n",
       " 'NC_021501.1 Klebsiella michiganensis strain E718 plasmid pKOX_NDM1, complete sequence',\n",
       " 'NC_018107.1 Klebsiella michiganensis E718 plasmid pKOX_R1, complete sequence',\n",
       " 'NC_016948.1 Mycobacterium intracellulare MOTT-64, complete genome',\n",
       " 'NC_016946.1 Mycobacterium intracellulare ATCC 13950, complete genome',\n",
       " 'NC_016947.1 Mycobacterium intracellulare MOTT-02, complete genome',\n",
       " 'NC_017049.1 Rickettsia prowazekii str. Chernikova, complete genome',\n",
       " 'NZ_CP007066.1 Fusobacterium nucleatum subsp. vincentii 3_1_27 plasmid, complete sequence',\n",
       " 'NZ_CP007064.1 Fusobacterium nucleatum subsp. vincentii 3_1_27 chromosome 1, complete sequence',\n",
       " 'NZ_CP007065.1 Fusobacterium nucleatum subsp. vincentii 3_1_27 chromosome 2, complete sequence',\n",
       " 'NZ_CP008742.1 Pseudomonas savastanoi pv. savastanoi NCPPB 3335, complete genome',\n",
       " 'NC_015678.1 Streptococcus parasanguinis ATCC 15912, complete genome',\n",
       " 'NC_014643.1 Rothia dentocariosa ATCC 17931, complete genome',\n",
       " 'NC_014618.1 Enterobacter lignolyticus SCF1, complete genome',\n",
       " 'NC_014625.1 Ketogulonicigenium vulgare Y25, complete genome',\n",
       " 'NC_014621.1 Ketogulonicigenium vulgare Y25 plasmid pYP1, complete sequence',\n",
       " 'NC_014626.1 Ketogulonicigenium vulgare Y25 plasmid pYP12, complete sequence',\n",
       " 'NC_017161.1 Hydrogenobacter thermophilus TK-6, complete genome',\n",
       " 'NC_014616.1 Bifidobacterium bifidum S17, complete genome',\n",
       " 'NC_014622.2 Paenibacillus polymyxa SC2, complete genome',\n",
       " 'NC_014628.2 Paenibacillus polymyxa SC2 plasmid pSC2, complete sequence',\n",
       " 'NC_017455.1 Halanaerobium praevalens DSM 2228, complete genome',\n",
       " 'NC_014623.1 Stigmatella aurantiaca DW4/3-1, complete genome',\n",
       " 'NC_014632.1 Ilyobacter polytropus DSM 2926, complete genome',\n",
       " 'NC_014633.1 Ilyobacter polytropus DSM 2926 plasmid pILYOP01, complete sequence',\n",
       " 'NC_014634.1 Ilyobacter polytropus DSM 2926 plasmid pILYOP02, complete sequence',\n",
       " 'NC_017451.1 Haemophilus influenzae R2866, complete genome',\n",
       " 'NC_017452.1 Haemophilus influenzae R2846, complete genome',\n",
       " 'NC_015174.1 Rubinisphaera brasiliensis DSM 5305, complete genome',\n",
       " 'NC_015214.1 Lactobacillus amylovorus strain 30SC, complete genome',\n",
       " 'NC_015213.1 Lactobacillus amylovorus strain 30SC plasmid pRKC30SC1, complete sequence',\n",
       " 'NC_015218.1 Lactobacillus amylovorus strain 30SC plasmid pRKC30SC2, complete sequence',\n",
       " 'NC_017569.1 Riemerella anatipestifer RA-GD, complete genome',\n",
       " 'NC_017295.1 Clostridium acetobutylicum EA 2018, complete genome',\n",
       " 'NC_017296.1 Clostridium acetobutylicum EA 2018 EA2018plasmid, complete sequence',\n",
       " 'NC_015470.1 Chlamydia psittaci 6BC, complete genome',\n",
       " 'NC_015217.1 Chlamydia psittaci 6BC plasmid p6BC, complete sequence',\n",
       " 'NZ_CP019725.1 Xanthomonas perforans 91-118 chromosome, complete genome',\n",
       " 'NC_015224.1 Yersinia enterocolitica subsp. palearctica 105.5R(r), complete genome',\n",
       " 'NC_015475.1 Yersinia enterocolitica subsp. palearctica 105.5R(r) plasmid 105.5R(r)p, complete sequence',\n",
       " 'NC_017374.1 Helicobacter pylori 2017, complete genome',\n",
       " 'NC_017381.1 Helicobacter pylori 2018, complete genome',\n",
       " 'NZ_CP017125.1 Mycoplasma capricolum subsp. capripneumoniae M1601 chromosome, complete genome',\n",
       " 'NC_015508.1 Agrobacterium sp. H13-3 linear chromosome, complete sequence',\n",
       " 'NC_015183.1 Agrobacterium sp. H13-3 circular chromosome, complete sequence',\n",
       " 'NC_015184.1 Agrobacterium sp. H13-3 plasmid pAspH13-3a, complete sequence',\n",
       " 'NC_017486.1 Lactococcus lactis subsp. lactis CV56, complete genome',\n",
       " 'NC_017483.1 Lactococcus lactis subsp. lactis CV56 plasmid pCV56A, complete sequence',\n",
       " 'NC_017487.1 Lactococcus lactis subsp. lactis CV56 plasmid pCV56B, complete sequence',\n",
       " 'NC_017484.1 Lactococcus lactis subsp. lactis CV56 plasmid pCV56C, complete sequence',\n",
       " 'NC_017485.1 Lactococcus lactis subsp. lactis CV56 plasmid pCV56D, complete sequence',\n",
       " 'NC_017488.1 Lactococcus lactis subsp. lactis CV56 plasmid pCV56E, complete sequence',\n",
       " 'NC_017244.1 Brucella melitensis M28 chromosome 1, complete sequence',\n",
       " 'NC_017245.1 Brucella melitensis M28 chromosome 2, complete sequence',\n",
       " 'NC_015259.1 Polymorphum gilvum SL003B-26A1, complete genome',\n",
       " 'NC_015258.1 Polymorphum gilvum SL003B-26A1 plasmid pSL003B, complete sequence',\n",
       " 'NC_015277.1 Sphingobacterium sp. 21, complete genome',\n",
       " 'NC_016783.1 Corynebacterium diphtheriae INCA 402, complete genome',\n",
       " 'NC_016800.1 Corynebacterium diphtheriae BH8, complete genome',\n",
       " 'NC_017033.1 Frateuria aurantia DSM 6220, complete genome',\n",
       " 'NC_019960.1 Prevotella dentalis DSM 3688 chromosome 1, complete sequence',\n",
       " 'NC_019968.1 Prevotella dentalis DSM 3688 chromosome 2, complete sequence',\n",
       " 'NC_019961.1 Prevotella dentalis DSM 3688 plasmid pPREDE01, complete sequence',\n",
       " 'NC_019969.1 Prevotella dentalis DSM 3688 plasmid pPREDE02, complete sequence',\n",
       " 'NC_019892.1 Singulisphaera acidiphila DSM 18658, complete genome',\n",
       " 'NC_019893.1 Singulisphaera acidiphila DSM 18658 plasmid pSINAC01, complete sequence',\n",
       " 'NC_019894.1 Singulisphaera acidiphila DSM 18658 plasmid pSINAC02, complete sequence',\n",
       " 'NC_019895.1 Singulisphaera acidiphila DSM 18658 plasmid pSINAC03, complete sequence',\n",
       " 'NC_017098.1 Spirochaeta africana DSM 8902, complete genome',\n",
       " 'NC_017770.1 Solitalea canadensis DSM 3403, complete genome',\n",
       " 'NC_016787.1 Corynebacterium diphtheriae HC03, complete genome',\n",
       " 'NC_020272.1 Bacillus amyloliquefaciens IT-45, complete genome',\n",
       " 'NC_020273.1 Bacillus amyloliquefaciens IT-45 plasmid pBA45-1, whole genome shotgun sequence',\n",
       " 'NC_021171.1 Bacillus sp. 1NLA3E, complete genome',\n",
       " 'NZ_CP007053.1 Opitutaceae bacterium TAV5, complete genome',\n",
       " 'NZ_CP007054.1 Opitutaceae bacterium TAV5 plasmid, complete sequence',\n",
       " 'NC_016745.1 Oceanimonas sp. GK1, complete genome',\n",
       " 'NC_016746.1 Oceanimonas sp. GK1 plasmid pOCEGK01, complete sequence',\n",
       " 'NC_016747.1 Oceanimonas sp. GK1 plasmid pOCEGK02, complete sequence',\n",
       " 'NZ_CP007035.1 Niabella soli DSM 19437, complete genome',\n",
       " 'NC_019903.1 Desulfitobacterium dichloroeliminans LMG P-21439, complete genome',\n",
       " 'NC_018017.1 Desulfitobacterium dehalogenans ATCC 51507, complete genome',\n",
       " 'NC_016791.1 Clostridium sp. BNL1100, complete genome',\n",
       " 'NZ_CP006932.1 Candidatus Hepatoplasma crinochetorum Av, complete genome',\n",
       " 'NZ_CP006829.1 Thalassolituus oleivorans R6-15, complete genome',\n",
       " 'NZ_CP007122.1 Lactobacillus paracasei N1115, complete genome',\n",
       " 'NZ_CP007123.1 Lactobacillus paracasei N1115 plasmid, complete sequence',\n",
       " 'NZ_CP007124.1 Lactobacillus paracasei N1115 plasmid, complete sequence',\n",
       " 'NZ_CP007125.1 Lactobacillus paracasei N1115 plasmid, complete sequence',\n",
       " 'NZ_CP007126.1 Lactobacillus paracasei N1115 plasmid, complete sequence',\n",
       " 'NC_022568.1 Listeria monocytogenes EGD, complete genome',\n",
       " 'NZ_CP007244.1 Bacillus amyloliquefaciens subsp. plantarum TrigoCor1448, complete genome',\n",
       " 'NZ_CP006742.1 Bacillus anthracis str. SVA11, complete genome',\n",
       " 'NZ_CP006743.1 Bacillus anthracis str. SVA11 plasmid pXO1, complete sequence',\n",
       " 'NZ_CP006744.1 Bacillus anthracis str. SVA11 plasmid pXO2, complete sequence',\n",
       " 'NZ_CP007183.1 Campylobacter coli RM1875, complete genome',\n",
       " 'NZ_CP007185.1 Campylobacter coli RM1875 plasmid pRM1875_2.4kb, complete sequence',\n",
       " 'NZ_CP007186.1 Campylobacter coli RM1875 plasmid pRM1875_3.3kb, complete sequence',\n",
       " 'NZ_CP007187.1 Campylobacter coli RM1875 plasmid pRM1875_3.4kbp, complete sequence',\n",
       " 'NZ_CP007184.1 Campylobacter coli RM1875 plasmid pRM1875_35kb, complete sequence',\n",
       " 'NZ_CP007181.1 Campylobacter coli RM4661, complete genome',\n",
       " 'NZ_CP007182.1 Campylobacter coli RM4661 plasmid pRM4661_48kbp, complete sequence',\n",
       " 'NZ_CP007179.1 Campylobacter coli RM5611, complete genome',\n",
       " 'NZ_CP007180.1 Campylobacter coli RM5611 plasmid pRM5611_48kb, complete sequence',\n",
       " 'NZ_CP004368.1 Burkholderia pseudomallei MSHR520 chromosome 1, complete sequence',\n",
       " 'NZ_CP004369.1 Burkholderia pseudomallei MSHR520 chromosome 2, complete sequence',\n",
       " 'NZ_CP004373.1 Gluconobacter oxydans DSM 3504, complete genome',\n",
       " 'NZ_CP006571.1 Chlamydia avium 10DC88, complete genome',\n",
       " 'NZ_CP006572.1 Chlamydia avium 10DC88 plasmid p10DC88, complete sequence',\n",
       " 'NZ_CP007410.1 Pseudomonas brassicacearum strain DF41, complete genome',\n",
       " 'NZ_CP007436.1 Bacillus pumilus strain MTCC B6033, complete genome',\n",
       " 'NZ_CP007441.1 Pseudomonas stutzeri strain 28a24, complete genome',\n",
       " 'NC_022904.2 Pandoraea pnomenusa 3kgm, complete genome',\n",
       " 'NC_022198.1 Corynebacterium argentoratense DSM 44202, complete genome',\n",
       " 'NC_022117.1 Chlamydia trachomatis strain F/1-93, complete genome',\n",
       " 'NC_022118.1 Chlamydia trachomatis strain F/6-94, complete genome',\n",
       " 'NC_022873.1 Bacillus thuringiensis YBT-1518, complete genome',\n",
       " 'NC_020124.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0228, complete sequence',\n",
       " 'NC_022874.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0229, complete sequence',\n",
       " 'NC_022875.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0230, complete sequence',\n",
       " 'NC_022876.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0231, complete sequence',\n",
       " 'NC_022877.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0232, complete sequence',\n",
       " 'NC_022882.1 Bacillus thuringiensis YBT-1518 plasmid pBMB0233, complete sequence',\n",
       " 'NC_022886.1 Helicobacter pylori BM012A, complete genome',\n",
       " 'NC_022911.1 Helicobacter pylori BM012S, complete genome',\n",
       " 'NZ_CP017152.1 Acinetobacter baumannii DU202, complete genome',\n",
       " 'NZ_CP010979.1 Pseudomonas putida S13.1.2, complete genome',\n",
       " 'NC_022900.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023, complete genome',\n",
       " 'NC_022901.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023 plasmid, complete sequence',\n",
       " 'NC_022913.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023 plasmid, complete sequence',\n",
       " 'NC_022902.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023 plasmid, complete sequence',\n",
       " 'NC_022903.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023 plasmid, complete sequence',\n",
       " 'NC_022910.1 Zymomonas mobilis subsp. mobilis str. CP4 = NRRL B-14023 plasmid, complete sequence',\n",
       " 'NC_022909.1 Lactobacillus johnsonii N6.2, complete genome',\n",
       " 'NZ_CP009365.1 Pseudomonas soli strain SJ10, complete genome',\n",
       " 'NC_022964.1 Dehalococcoides mccartyi GY50, complete genome',\n",
       " 'NZ_HG738867.1 Escherichia coli str. K-12 substr. MC4100 complete genome',\n",
       " 'NC_022535.1 Rhizobium sp. IRBG74 circular chromosome, complete genome',\n",
       " 'NC_022545.1 Rhizobium sp. IRBG74 linear chromosome, complete genome',\n",
       " 'NC_022536.1 Rhizobium sp. IRBG74 plasmid IRBL74_p, complete sequence',\n",
       " 'NZ_LN794217.1 Rickettsia monacensis, strain IrR/Munich, complete genome',\n",
       " 'NZ_CP011913.1 Corynebacterium ulcerans FRC58, complete genome',\n",
       " 'NC_019896.1 Bacillus subtilis subsp. subtilis str. BSP1, complete genome',\n",
       " 'NC_019902.2 Thioalkalivibrio nitratireducens DSM 14787, complete genome',\n",
       " 'NC_019908.1 Brachyspira pilosicoli P43/6/78, complete genome',\n",
       " 'NC_019904.1 Echinicola vietnamensis DSM 17526, complete genome',\n",
       " 'NC_019905.1 Pseudomonas putida HB3267, complete genome',\n",
       " 'NC_019906.1 Pseudomonas putida HB3267 plasmid pPC9, complete sequence',\n",
       " 'NC_019907.1 Liberibacter crescens BT-1, complete genome',\n",
       " 'NC_019940.1 Thioflavicoccus mobilis 8321, complete genome',\n",
       " 'NC_019941.1 Thioflavicoccus mobilis 8321 plasmid pTHIMO01, complete sequence',\n",
       " 'NC_019936.1 Pseudomonas stutzeri RCH2, complete genome',\n",
       " 'NC_019937.1 Pseudomonas stutzeri RCH2 plasmid pPSEST01, complete sequence',\n",
       " 'NC_019938.1 Pseudomonas stutzeri RCH2 plasmid pPSEST02, complete sequence',\n",
       " 'NC_019939.1 Pseudomonas stutzeri RCH2 plasmid pPSEST03, complete sequence',\n",
       " 'NC_019955.1 Vibrio parahaemolyticus BB22OP chromosome 1, complete sequence',\n",
       " 'NC_019971.1 Vibrio parahaemolyticus BB22OP chromosome 2, complete sequence',\n",
       " 'NC_019970.1 Thermoanaerobacterium thermosaccharolyticum M0795, complete genome',\n",
       " 'NC_019956.1 Thermoanaerobacterium thermosaccharolyticum M0795 plasmid pTHETHE01, complete sequence',\n",
       " 'NC_019966.1 Mycobacterium sp. JS623, complete genome',\n",
       " 'NC_019957.1 Mycobacterium sp. JS623 plasmid pMYCSM01, complete sequence',\n",
       " 'NC_019958.1 Mycobacterium sp. JS623 plasmid pMYCSM02, complete sequence',\n",
       " 'NC_019959.1 Mycobacterium sp. JS623 plasmid pMYCSM03, complete sequence',\n",
       " 'NC_019978.1 Halobacteroides halobius DSM 5150, complete genome',\n",
       " 'NC_019673.1 Saccharothrix espanaensis DSM 44229 complete genome',\n",
       " 'NC_019949.1 Mycoplasma cynos C142 complete genome',\n",
       " 'NC_019954.2 Tepidanaerobacter acetatoxydans Re1 complete genome',\n",
       " 'NZ_CP007528.1 Salmonella enterica subsp. enterica serovar Enteritidis str. CDC_2010K_0968, complete genome',\n",
       " 'NZ_CP007529.1 Salmonella enterica subsp. enterica serovar Enteritidis str. CDC_2010K_0968 plasmid00, complete sequence',\n",
       " 'NZ_CP012347.1 Salmonella enterica subsp. enterica serovar Pullorum str. ATCC 9120, complete genome',\n",
       " 'NZ_CP012348.1 Salmonella enterica subsp. enterica serovar Pullorum str. ATCC 9120 plasmid pCFSAN000725_01, complete sequence',\n",
       " 'NZ_CP007255.1 Rhodococcus erythropolis R138, complete genome',\n",
       " 'NZ_CP007256.1 Rhodococcus erythropolis R138 plasmid pCRE138, complete sequence',\n",
       " 'NZ_CP007257.1 Rhodococcus erythropolis R138 plasmid pLRE138, complete sequence',\n",
       " 'NZ_CP015880.1 Ensifer adhaerens strain Casida A chromosome, complete genome',\n",
       " 'NZ_CP015881.1 Ensifer adhaerens strain Casida A plasmid pCasidaAA, complete sequence',\n",
       " 'NZ_CP015882.1 Ensifer adhaerens strain Casida A plasmid pCasidaAB, complete sequence',\n",
       " 'NZ_CP007691.1 Brucella suis bv. 2 strain PT09143 chromosome I, complete sequence',\n",
       " 'NZ_CP007692.1 Brucella suis bv. 2 strain PT09143 chromosome II, complete sequence',\n",
       " 'NZ_CP007693.1 Brucella suis bv. 2 strain PT09172 chromosome I, complete sequence',\n",
       " 'NZ_CP007694.1 Brucella suis bv. 2 strain PT09172 chromosome II, complete sequence',\n",
       " 'NZ_CP007697.1 Brucella suis bv. 2 strain Bs364CITA chromosome I, complete sequence',\n",
       " 'NZ_CP007698.1 Brucella suis bv. 2 strain Bs364CITA chromosome II, complete sequence',\n",
       " 'NZ_CP007720.1 Brucella suis bv. 2 strain Bs396CITA chromosome I, complete sequence',\n",
       " 'NZ_CP007721.1 Brucella suis bv. 2 strain Bs396CITA chromosome II, complete sequence',\n",
       " 'NZ_CP007805.1 Haemophilus influenzae CGSHiCZ412602, complete genome',\n",
       " 'NZ_CP007803.1 Mycobacterium tuberculosis K, complete genome',\n",
       " 'NZ_CP006696.1 Xylella fastidiosa subsp. sandyi Ann-1, complete genome',\n",
       " 'NZ_CP006697.1 Xylella fastidiosa subsp. sandyi Ann-1 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP006740.1 Xylella fastidiosa MUL0034, complete genome',\n",
       " 'NZ_CP006739.1 Xylella fastidiosa MUL0034 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP006986.1 Rhizobium sp. IE4771 chromosome, complete genome',\n",
       " 'NZ_CP006987.1 Rhizobium sp. IE4771 plasmid pRetIE4771a, complete sequence',\n",
       " 'NZ_CP006988.1 Rhizobium sp. IE4771 plasmid pRetIE4771b, complete sequence',\n",
       " 'NZ_CP006989.1 Rhizobium sp. IE4771 plasmid pRetIE4771c, complete sequence',\n",
       " 'NZ_CP006990.1 Rhizobium sp. IE4771 plasmid pRetIE4771d, complete sequence',\n",
       " 'NZ_CP006991.1 Rhizobium sp. IE4771 plasmid pRetIE4771e, complete sequence',\n",
       " 'NZ_CP008696.1 Pseudomonas chlororaphis strain PA23, complete genome',\n",
       " 'NZ_CP006819.1 Streptococcus thermophilus ASCC 1275, complete genome',\n",
       " 'NZ_CP007695.1 Brucella suis bv. 2 strain Bs143CITA chromosome I, complete sequence',\n",
       " 'NZ_CP007696.1 Brucella suis bv. 2 strain Bs143CITA chromosome II, complete sequence',\n",
       " 'NC_020533.1 Staphylococcus aureus subsp. aureus ST228 complete genome, isolate 16035',\n",
       " 'NC_020534.1 Staphylococcus aureus subsp. aureus ST228 plasmid pI4T8 complete sequence, isolate 16035',\n",
       " 'NC_020566.1 Staphylococcus aureus subsp. aureus ST228 complete genome, isolate 16125',\n",
       " 'NC_020535.1 Staphylococcus aureus subsp. aureus ST228 plasmid pI5S5 complete sequence, isolate 16125',\n",
       " 'NC_020568.1 Staphylococcus aureus subsp. aureus ST228 complete genome, isolate 18583',\n",
       " 'NC_020539.1 Staphylococcus aureus subsp. aureus ST228 plasmid pI8T7 complete sequence, isolate 18583',\n",
       " 'NC_021173.1 Burkholderia thailandensis MSMB121 chromosome 1, complete sequence',\n",
       " 'NC_021174.1 Burkholderia thailandensis MSMB121 chromosome 2, complete sequence',\n",
       " 'NC_021176.1 Salmonella enterica subsp. enterica serovar Typhi str. Ty21a, complete genome',\n",
       " 'NC_021175.1 Streptococcus cristatus AS 1.3089, complete genome',\n",
       " 'NC_021177.1 Streptomyces fulvissimus DSM 40593, complete sequence',\n",
       " 'NZ_CP010539.1 Mycoplasma pneumoniae 19294, complete genome',\n",
       " 'NC_021182.1 Clostridium pasteurianum BC1, complete genome',\n",
       " 'NC_021183.1 Clostridium pasteurianum BC1 plasmid pCLOPA01, complete sequence',\n",
       " 'NC_021181.2 Lactobacillus acidophilus La-14, complete genome',\n",
       " 'NC_021194.1 Mycobacterium tuberculosis EAI5/NITR206, complete genome',\n",
       " 'NC_021191.1 Actinoplanes sp. N902-109, complete genome',\n",
       " 'NC_021200.1 Mycobacterium avium subsp. paratuberculosis MAP4, complete genome',\n",
       " 'NC_021213.1 Streptococcus suis TL13, complete genome',\n",
       " 'NZ_CP009144.1 Sinorhizobium meliloti strain RMO17, complete genome',\n",
       " 'NZ_CP009145.1 Sinorhizobium meliloti strain RMO17 plasmid pSymA, complete sequence',\n",
       " 'NZ_CP009146.1 Sinorhizobium meliloti strain RMO17 plasmid pSymB, complete sequence',\n",
       " 'NZ_CP009215.1 Corynebacterium ureicelerivorans strain IMMIB RIV-2301, complete genome',\n",
       " 'NZ_CP009216.1 Corynebacterium ureicelerivorans strain IMMIB RIV-2301 plasmid unnamed, complete sequence',\n",
       " 'NZ_CP008853.1 Bacillus anthracis strain Cvac02, complete genome',\n",
       " 'NZ_CP008855.1 Bacillus sp. X1(2014), complete genome',\n",
       " 'NZ_CP009278.1 Sphingobacterium sp. ML3W, complete genome',\n",
       " 'NZ_CP007607.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520, complete genome',\n",
       " 'NZ_CP007612.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB11, complete sequence',\n",
       " 'NZ_CP007608.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB2062, complete sequence',\n",
       " 'NZ_CP007615.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB293, complete sequence',\n",
       " 'NZ_CP007616.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB400, complete sequence',\n",
       " 'NZ_CP007613.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB69, complete sequence',\n",
       " 'NZ_CP007609.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB7921, complete sequence',\n",
       " 'NZ_CP007610.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB8240, complete sequence',\n",
       " 'NZ_CP007611.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB8513, complete sequence',\n",
       " 'NZ_CP007614.1 Bacillus thuringiensis serovar kurstaki str. YBT-1520 plasmid pBMB95, complete sequence',\n",
       " 'NZ_CP003424.1 Serratia sp. SCBI, complete genome',\n",
       " 'NZ_CP003425.1 Serratia sp. SCBI plasmid SCBI_Pl, complete sequence',\n",
       " 'NZ_CP006683.1 Melissococcus plutonius S1, complete genome',\n",
       " 'NZ_CP006684.1 Melissococcus plutonius S1 plasmid pMEPL_178, complete sequence',\n",
       " 'NZ_CP009239.1 Cellulophaga lytica strain HI1, complete genome',\n",
       " 'NZ_CP009089.1 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE7-100819, complete genome',\n",
       " 'NZ_CP008897.1 Enterobacter cloacae ECNIH3, complete genome',\n",
       " 'NZ_CP008898.1 Enterobacter cloacae ECNIH3 plasmid pENT-576, complete sequence',\n",
       " 'NZ_CP008899.1 Enterobacter cloacae ECNIH3 plasmid pENT-8a4, complete sequence',\n",
       " 'NZ_CP008900.1 Enterobacter cloacae ECNIH3 plasmid pENT-d4a, complete sequence',\n",
       " 'NZ_CP008901.1 Enterobacter cloacae ECNIH3 plasmid pKPC-47e, complete sequence',\n",
       " 'NZ_CP009090.1 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE8-1021710, complete genome',\n",
       " 'NZ_CP008905.1 Enterobacter cloacae ECR091, complete genome',\n",
       " 'NZ_CP008906.1 Enterobacter cloacae ECR091 plasmid pENT-08e, complete sequence',\n",
       " 'NZ_CP008907.1 Enterobacter cloacae ECR091 plasmid pENT-4bd, complete sequence',\n",
       " 'NZ_CP008908.1 Enterobacter cloacae ECR091 plasmid pKPC-47e, complete sequence',\n",
       " 'NZ_CP009091.1 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE9-10012, complete genome',\n",
       " 'NZ_CP008943.1 Yersinia pseudotuberculosis strain ATCC 6904, complete genome',\n",
       " 'NZ_CP009092.1 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE10-10052, complete genome',\n",
       " 'NZ_CP009093.1 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE11-10058, complete genome',\n",
       " 'NZ_CP009085.2 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE3-98983-4, complete genome',\n",
       " 'NZ_CP009084.2 Salmonella enterica subsp. enterica serovar Enteritidis strain OLF-SE2-98984-6, complete genome',\n",
       " 'NZ_CP007229.1 Mycoplasma dispar strain ATCC 27140, complete genome',\n",
       " 'NZ_CP016174.1 Amycolatopsis orientalis strain B-37, complete genome',\n",
       " 'NZ_CP010954.1 Sphingobium sp. YBL2, complete genome',\n",
       " 'NZ_CP010955.1 Sphingobium sp. YBL2 plasmid 1pYBL2-1, complete sequence',\n",
       " 'NZ_CP010956.1 Sphingobium sp. YBL2 plasmid 2pYBL2-2, complete sequence',\n",
       " 'NZ_CP010957.1 Sphingobium sp. YBL2 plasmid 3pYBL2-3, complete sequence',\n",
       " 'NZ_CP010958.1 Sphingobium sp. YBL2 plasmid 4pYBL2-4, complete sequence',\n",
       " 'NZ_CP010959.1 Sphingobium sp. YBL2 plasmid 5pYBL2-5, complete sequence',\n",
       " 'NZ_CP010960.1 Sphingobium sp. YBL2 plasmid 6pYBL2-6, complete sequence',\n",
       " 'NZ_CP009677.1 Mycoplasma hominis strain AF1, complete genome',\n",
       " 'NZ_CP007166.1 Xanthomonas oryzae pv. oryzae PXO86, complete genome',\n",
       " 'NZ_CP010976.1 Paenibacillus sp. IHBB 10380, complete genome',\n",
       " 'NZ_CP010977.1 Paenibacillus sp. IHBB 10380 plasmid, complete sequence',\n",
       " 'NZ_CP011005.1 Arthrobacter sp. IHBB 11108, complete genome',\n",
       " 'NZ_CP011006.1 Arthrobacter sp. IHBB 11108 plasmid pAG001, complete sequence',\n",
       " 'NZ_CP011004.1 Listeria monocytogenes strain N2306, complete genome',\n",
       " 'NZ_CP007699.2 Streptomyces lydicus strain A02 chromosome, complete genome',\n",
       " 'NZ_LN680001.1 Bacillus sp. BS34A genome assembly, chromosome: I',\n",
       " 'NZ_LM995445.1 Mycoplasma capricolum subsp. capripneumoniae genome assembly 9231-Abomsa assembly V8, chromosome : I',\n",
       " 'NZ_LN554882.1 Chlamydophila abortus genome assembly CAAB7, chromosome : 1',\n",
       " 'NZ_LM995446.1 Escherichia coli genome assembly EcRV308Chr, chromosome : 1',\n",
       " 'NZ_LM997412.1 Peptoniphilus sp. ING2-D1G genome assembly, chromosome: I',\n",
       " 'NZ_LN794158.1 Candidatus Methylopumilus turicensis genome assembly Candidatus Methylopumilus turicensis MMS-10A-171, chromosome : 1',\n",
       " 'NZ_LM651928.1 Shigella flexneri genome assembly NCTC1, chromosome : 1',\n",
       " 'NZ_LK936442.1 Burkholderia pseudomallei genome assembly BP_3921g, chromosome : 1',\n",
       " 'NZ_LK936443.1 Burkholderia pseudomallei genome assembly BP_3921g, chromosome : 2',\n",
       " 'NZ_LN614827.1 Legionella fallonii LLAP-10 genome assembly LFA, chromosome : I',\n",
       " 'NZ_LN614828.1 Legionella fallonii LLAP-10 genome assembly LFA, plasmid : II',\n",
       " 'NZ_LN614829.1 Legionella fallonii LLAP-10 genome assembly LFA, plasmid : III',\n",
       " 'NZ_CP007686.1 Listeria monocytogenes strain L2624, complete genome',\n",
       " 'NZ_CP007685.1 Listeria monocytogenes strain L2676, complete genome',\n",
       " 'NZ_CP011511.1 Shigella boydii strain ATCC 9210, complete genome',\n",
       " 'NZ_CP007684.1 Listeria monocytogenes strain L2626, complete genome',\n",
       " 'NZ_CP011501.1 Pandoraea apista strain AU2161, complete genome',\n",
       " 'NZ_CP011509.1 Archangium gephyra strain DSM 2261, complete genome',\n",
       " 'NZ_CP009498.1 Endomicrobium proavitum strain Rsa215, complete genome',\n",
       " 'NZ_CP011827.2 Xanthomonas citri pv. citri strain jx-6, complete genome',\n",
       " 'NZ_CP013665.1 Xanthomonas citri pv. citri strain jx-6 plasmid pXAC33, complete sequence',\n",
       " 'NZ_CP013664.1 Xanthomonas citri pv. citri strain jx-6 plasmid pXAC64, complete sequence',\n",
       " 'NZ_CP011805.1 Altererythrobacter marensis strain KCTC 22370, complete genome',\n",
       " 'NZ_CP007601.1 Staphylococcus capitis subsp. capitis strain AYP1020, complete genome',\n",
       " 'NZ_CP007602.1 Staphylococcus capitis subsp. capitis strain AYP1020 plasmid pAYP1020, complete sequence',\n",
       " 'NZ_CP011503.1 Burkholderia pyrrocinia strain DSM 10685 chromosome 1, complete sequence',\n",
       " 'NZ_CP011504.1 Burkholderia pyrrocinia strain DSM 10685 chromosome 2, complete sequence',\n",
       " 'NZ_CP011505.1 Burkholderia pyrrocinia strain DSM 10685 chromosome 3, complete sequence',\n",
       " 'NZ_CP011506.1 Burkholderia pyrrocinia strain DSM 10685 plasmid p2327, complete sequence',\n",
       " 'NZ_CP011770.1 Croceicoccus naphthovorans strain PQ-2, complete genome',\n",
       " 'NZ_CP011771.1 Croceicoccus naphthovorans strain PQ-2 plasmid p1, complete sequence',\n",
       " 'NZ_CP011772.1 Croceicoccus naphthovorans strain PQ-2 plasmid p2, complete sequence',\n",
       " 'NZ_CP011807.3 Pandoraea faecigallinarum strain DSM 23572, complete genome',\n",
       " 'NZ_CP011808.2 Pandoraea faecigallinarum strain DSM 23572 plasmid pPF72-1, complete sequence',\n",
       " 'NZ_CP011809.2 Pandoraea faecigallinarum strain DSM 23572 plasmid pPF72-2, complete sequence',\n",
       " 'NZ_CP006636.1 Escherichia coli PCN061, complete genome',\n",
       " 'NZ_CP006637.1 Escherichia coli PCN061 plasmid PCN061p1, complete sequence',\n",
       " 'NZ_CP006638.1 Escherichia coli PCN061 plasmid PCN061p2, complete sequence',\n",
       " 'NZ_CP006639.1 Escherichia coli PCN061 plasmid PCN061p3, complete sequence',\n",
       " 'NZ_CP006640.1 Escherichia coli PCN061 plasmid PCN061p4, complete sequence',\n",
       " 'NZ_CP006641.1 Escherichia coli PCN061 plasmid PCN061p5, complete sequence',\n",
       " 'NZ_CP006642.1 Escherichia coli PCN061 plasmid PCN061p6, complete sequence',\n",
       " 'NZ_CP009743.1 Burkholderia contaminans strain MS14 chromosome 1, complete sequence',\n",
       " 'NZ_CP009744.1 Burkholderia contaminans strain MS14 chromosome 2, complete sequence',\n",
       " 'NZ_CP009745.1 Burkholderia contaminans strain MS14 chromosome 3, complete sequence',\n",
       " 'NZ_CP011855.1 Spiroplasma atrichopogonis strain GNAT3597, complete genome',\n",
       " 'NZ_CP011856.1 Spiroplasma eriocheiris strain DSM 21848, complete genome',\n",
       " 'NZ_CP011798.1 Enterobacter cloacae strain UW5, complete genome',\n",
       " 'NZ_CP007667.1 Neisseria meningitidis strain B6116/77, complete genome',\n",
       " 'NZ_CP007668.1 Neisseria meningitidis M0579, complete genome',\n",
       " 'NZ_CP011882.1 Bacillus subtilis strain TO-A JPC, complete genome',\n",
       " 'NZ_CP011804.1 Clostridium carboxidivorans P7 plasmid, complete sequence',\n",
       " 'NZ_CP011803.1 Clostridium carboxidivorans P7, complete genome',\n",
       " 'NZ_CP005969.1 Pseudomonas syringae pv. syringae B301D, complete genome',\n",
       " 'NZ_CP007584.2 Salmonella enterica subsp. enterica serovar Anatum str. USDA-ARS-USMARC-1735, complete genome',\n",
       " 'NZ_CP014707.1 Salmonella enterica subsp. enterica serovar Anatum strain USMARC-1735 plasmid pSAN1-1735, complete sequence',\n",
       " 'NZ_CP006003.1 Myxococcus fulvus 124B02, complete genome',\n",
       " 'NZ_CP011374.1 Moraxella bovoculi strain 58069, complete genome',\n",
       " 'NZ_CP011375.1 Moraxella bovoculi strain 58069 plasmid, complete sequence',\n",
       " 'NZ_CP011376.1 Moraxella bovoculi strain 22581, complete genome',\n",
       " 'NZ_CP011377.1 Moraxella bovoculi strain 23343, complete genome',\n",
       " 'NZ_CP011378.1 Moraxella bovoculi strain 28389, complete genome',\n",
       " 'NZ_CP011379.1 Moraxella bovoculi strain 33362, complete genome',\n",
       " 'NZ_CP011380.2 Moraxella bovoculi strain 57922, complete genome',\n",
       " 'NZ_CP011365.1 Salmonella enterica subsp. enterica serovar Typhimurium strain FORC_015, complete genome',\n",
       " 'NZ_CP007497.1 Streptococcus suis strain ZY05719, complete genome',\n",
       " 'NZ_CP007537.1 Streptococcus pyogenes strain AP1, complete genome',\n",
       " 'NZ_CP009922.2 Streptomyces xiamenensis strain 318, complete genome',\n",
       " 'NZ_CP011114.1 Paenibacillus durus ATCC 35681, complete genome',\n",
       " 'NZ_CP011366.1 Salinicoccus halodurans strain H3B36, complete genome',\n",
       " 'NZ_CP011398.2 Listeria monocytogenes strain CFSAN008100 chromosome, complete genome',\n",
       " 'NZ_CP011399.1 Listeria monocytogenes strain CFSAN008100 plasmid pCFSAN008100, complete sequence',\n",
       " 'NZ_AP012555.1 Mycobacterium avium subsp. hominissuis TH135 chromosomal DNA, complete genome',\n",
       " 'NZ_AP012556.1 Mycobacterium avium subsp. hominissuis TH135 plasmid pMAH135 DNA, complete genome',\n",
       " 'NZ_AP013354.1 Helicobacter pylori 26695-1 DNA, complete genome',\n",
       " 'NZ_AP013356.1 Helicobacter pylori 26695-1CL DNA, complete genome',\n",
       " 'NZ_AP013355.1 Helicobacter pylori 26695-1CH DNA, complete genome',\n",
       " 'NZ_AP013293.1 Candidatus Sulcia muelleri PSPU DNA, complete genome',\n",
       " 'NZ_AP014582.1 Bordetella bronchiseptica DNA, complete genome, strain: S798',\n",
       " 'NZ_AP013294.1 Bacillus sp. OxB-1 DNA, complete genome',\n",
       " 'NZ_AP014524.1 Vibrio cholerae MS6 DNA, complete genome, chromosome 1',\n",
       " 'NZ_AP014525.1 Vibrio cholerae MS6 DNA, complete genome, chromosome 2',\n",
       " 'NZ_AP012549.1 Cyanobacterium endosymbiont of Epithemia turgida isolate EtSB Lake Yunoko DNA, complete genome',\n",
       " 'NZ_AP014646.1 Pseudomonas aeruginosa DNA, complete genome, strain: NCGM 1984',\n",
       " 'NZ_AP014622.1 Pseudomonas aeruginosa DNA, complete genome, strain: NCGM 1900',\n",
       " 'NZ_AP014658.1 Bifidobacterium longum DNA, complete genome, strain: 105-A',\n",
       " 'NZ_AP013028.1 Wolbachia endosymbiont of Cimex lectularius DNA, complete genome',\n",
       " 'NZ_CP026055.1 Aeromonas caviae strain FDAARGOS_72 chromosome, complete genome',\n",
       " 'NZ_CP026052.1 Salmonella enterica strain FDAARGOS_70 chromosome, complete genome',\n",
       " 'NZ_CP026053.1 Salmonella enterica strain FDAARGOS_70 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP026054.1 Salmonella enterica strain FDAARGOS_70 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP026051.1 Proteus mirabilis strain FDAARGOS_67 chromosome, complete genome',\n",
       " 'NZ_CP026050.1 Serratia marcescens strain FDAARGOS_65 chromosome, complete genome',\n",
       " 'NZ_CP026047.1 Raoultella planticola strain FDAARGOS_64 chromosome, complete genome',\n",
       " 'NZ_CP026048.1 Raoultella planticola strain FDAARGOS_64 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP026049.1 Raoultella planticola strain FDAARGOS_64 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP026046.1 Morganella morganii strain FDAARGOS_63 chromosome, complete genome',\n",
       " 'NZ_CP026045.1 Citrobacter freundii strain FDAARGOS_61 chromosome, complete genome',\n",
       " 'NZ_CP026044.1 Proteus mirabilis strain FDAARGOS_60 chromosome, complete genome',\n",
       " 'NZ_CP009850.1 Enterobacter cloacae strain ECNIH4, complete genome',\n",
       " 'NZ_CP009851.1 Enterobacter cloacae strain ECNIH4 plasmid pENT-c88, complete sequence',\n",
       " 'NZ_CP009852.1 Enterobacter cloacae strain ECNIH4 plasmid pENT-e56, complete sequence',\n",
       " 'NZ_CP009853.1 Enterobacter cloacae strain ECNIH4 plasmid pKPC-860, complete sequence',\n",
       " 'NZ_CP009880.1 Pantoea sp. PSNIH1, complete genome',\n",
       " 'NZ_CP009881.1 Pantoea sp. PSNIH1 plasmid pKPC-1c5, complete sequence',\n",
       " 'NZ_CP010325.1 Pantoea sp. PSNIH1 plasmid pPSP-057, complete sequence',\n",
       " 'NZ_CP009882.1 Pantoea sp. PSNIH1 plasmid pPSP-26e, complete sequence',\n",
       " 'NZ_CP010326.1 Pantoea sp. PSNIH1 plasmid pPSP-3a9, complete sequence',\n",
       " 'NZ_CP009883.1 Pantoea sp. PSNIH1 plasmid pPSP-a3e, complete sequence',\n",
       " 'NZ_CP009884.1 Pantoea sp. PSNIH1 plasmid pPSP-ee2, complete sequence',\n",
       " 'NZ_CP009854.1 Enterobacter cloacae strain ECNIH5, complete genome',\n",
       " 'NZ_CP009855.1 Enterobacter cloacae strain ECNIH5 plasmid pENT-22e, complete sequence',\n",
       " 'NZ_CP009856.1 Enterobacter cloacae strain ECNIH5 plasmid pENT-784, complete sequence',\n",
       " 'NZ_CP009857.1 Enterobacter cloacae strain ECNIH5 plasmid pENT-d0d, complete sequence',\n",
       " 'NZ_CP009858.1 Enterobacter cloacae strain ECNIH5 plasmid pKPC-47e, complete sequence',\n",
       " 'NZ_CP009859.1 Escherichia coli strain ECONIH1, complete genome',\n",
       " 'NZ_CP009860.1 Escherichia coli strain ECONIH1 plasmid pECO-824, complete sequence',\n",
       " 'NZ_CP009861.1 Escherichia coli strain ECONIH1 plasmid pECO-b75, complete sequence',\n",
       " 'NZ_CP009862.1 Escherichia coli strain ECONIH1 plasmid pKPC-629, complete sequence',\n",
       " 'NZ_CP009863.1 Klebsiella pneumoniae subsp. pneumoniae strain KPNIH29, complete genome',\n",
       " 'NZ_CP009864.1 Klebsiella pneumoniae subsp. pneumoniae strain KPNIH29 plasmid pKPC-e4e, complete sequence',\n",
       " 'NZ_CP009865.1 Klebsiella pneumoniae subsp. pneumoniae strain KPNIH29 plasmid pKPN-80a, complete sequence',\n",
       " 'NZ_CP009866.1 Pantoea sp. PSNIH2, complete genome',\n",
       " 'NZ_CP009867.1 Pantoea sp. PSNIH2 plasmid pKPC-56a, complete sequence',\n",
       " 'NZ_CP009868.1 Pantoea sp. PSNIH2 plasmid pPSP-100, complete sequence',\n",
       " 'NZ_CP009869.1 Pantoea sp. PSNIH2 plasmid pPSP-75c, complete sequence',\n",
       " 'NZ_CP009870.1 Pantoea sp. PSNIH2 plasmid pPSP-b98, complete sequence',\n",
       " 'NZ_CP009871.1 Pantoea sp. PSNIH2 plasmid pPSP-cd6, complete sequence',\n",
       " 'NZ_CP012074.1 Prevotella fusca JCM 17724 strain W1435 chromosome 1, complete sequence',\n",
       " 'NZ_CP012075.1 Prevotella fusca JCM 17724 strain W1435 chromosome 2, complete sequence',\n",
       " 'NZ_CP012067.1 Aggregatibacter aphrophilus strain W10433, complete genome',\n",
       " 'NZ_CP012072.1 Actinomyces meyeri strain W712, complete genome',\n",
       " 'NZ_CP012073.1 Ottowia sp. oral taxon 894 strain W10237, complete genome',\n",
       " 'NZ_CP012037.1 Francisella tularensis subsp. tularensis strain WY96, complete genome',\n",
       " 'NZ_CP012328.1 Spiroplasma turonicum strain Tab4c, complete genome',\n",
       " 'NZ_CP012332.1 Vulgatibacter incomptus strain DSM 27710, complete genome',\n",
       " 'NZ_CP011786.1 Bifidobacterium actinocoloniiforme DSM 22766, complete genome',\n",
       " 'NZ_CP011995.1 Porphyromonas gingivalis strain A7436, complete genome',\n",
       " 'NZ_CP011339.1 Microcystis panniformis FACHB-1757, complete genome',\n",
       " 'NZ_CP010319.1 Streptococcus agalactiae strain GBS85147, complete genome',\n",
       " 'NZ_CP012299.1 Microbacterium sp. CGR1, complete genome',\n",
       " 'NZ_CP011929.1 Marinobacter sp. CP1, complete genome',\n",
       " 'NZ_CP012357.1 Spiroplasma litorale strain TN-1, complete genome',\n",
       " 'NZ_CP012358.1 Oblitimonas alkaliphila strain B4199 chromosome, complete genome',\n",
       " 'NZ_CP012359.1 Oblitimonas alkaliphila strain C6819 chromosome, complete genome',\n",
       " 'NZ_CP012360.1 Oblitimonas alkaliphila strain C6918 chromosome, complete genome',\n",
       " 'NZ_CP012362.1 Oblitimonas alkaliphila strain D3318 chromosome, complete genome',\n",
       " 'NZ_CP012363.1 Oblitimonas alkaliphila strain E1086 chromosome, complete genome',\n",
       " 'NZ_CP014773.1 Mucilaginibacter sp. PAMC 26640 chromosome, complete genome',\n",
       " 'NZ_CP014772.1 Mucilaginibacter sp. PAMC 26640 plasmid unnamed, complete sequence',\n",
       " 'NZ_CP014771.1 Hymenobacter sp. PAMC 26554 chromosome, complete genome',\n",
       " 'NZ_CP014769.1 Hymenobacter sp. PAMC 26554 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP014770.1 Hymenobacter sp. PAMC 26554 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP014805.2 Elizabethkingia anophelis strain CSID_3015183678 chromosome, complete genome',\n",
       " 'NZ_CP014858.1 Bacillus subtilis subsp. subtilis strain D12-5 chromosome, complete genome',\n",
       " 'NZ_CP014783.1 Bacillus amyloliquefaciens strain B15 chromosome, complete genome',\n",
       " 'NZ_CP014790.1 Listeria monocytogenes strain 2015TE24968 chromosome, complete genome',\n",
       " 'NZ_CP015985.1 Listeria monocytogenes strain 2015TE24968 plasmid pl2015TE24968, complete sequence',\n",
       " 'NZ_CP014840.1 Bacillus subtilis subsp. globigii strain ATCC 49760 chromosome, complete genome',\n",
       " 'NZ_CP014784.1 Pseudomonas alcaligenes strain NEB 585, complete genome',\n",
       " 'NZ_CP014835.1 Streptococcus halotolerans strain HTS9 chromosome, complete genome',\n",
       " 'NZ_CP014856.1 Lysinibacillus sphaericus III(3)7, complete genome',\n",
       " 'NZ_CP014857.1 Lysinibacillus sphaericus III(3)7 plasmid, complete sequence',\n",
       " 'NZ_CP014847.1 Bacillus thuringiensis strain HD12, complete sequence',\n",
       " 'NZ_CP014848.1 Bacillus thuringiensis strain HD12 plasmid pHD120017, complete sequence',\n",
       " 'NZ_CP014849.1 Bacillus thuringiensis strain HD12 plasmid pHD120038, complete sequence',\n",
       " 'NZ_CP014850.1 Bacillus thuringiensis strain HD12 plasmid pHD120039, complete sequence',\n",
       " 'NZ_CP014851.1 Bacillus thuringiensis strain HD12 plasmid pHD120112, complete sequence',\n",
       " 'NZ_CP014852.1 Bacillus thuringiensis strain HD12 plasmid pHD120161, complete sequence',\n",
       " 'NZ_CP014853.1 Bacillus thuringiensis strain HD12 plasmid pHD120345, complete sequence',\n",
       " 'NZ_CP014949.1 Enterococcus faecalis strain LD33, complete genome',\n",
       " 'NZ_CP014991.1 Helicobacter himalayensis strain YS1, complete genome',\n",
       " 'NZ_CP014352.1 Acidipropionibacterium acidipropionici strain ATCC 55737 chromosome, complete genome',\n",
       " 'NZ_CP014353.1 Acidipropionibacterium acidipropionici strain ATCC 55737 plasmid unnamed, complete sequence',\n",
       " 'NZ_CP014867.1 Pseudomonas chlororaphis isolate 189 chromosome, complete genome',\n",
       " 'NZ_CP014861.1 Erysipelothrix rhusiopathiae strain GXBY-1 chromosome, complete genome',\n",
       " 'NZ_CP014865.1 Rickettsia prowazekii strain Naples-1 chromosome, complete genome',\n",
       " 'NZ_CP015004.1 Bacillus subtilis strain SZMC 6179J, complete genome',\n",
       " 'NZ_CP015005.1 Aminobacter aminovorans strain KCTC 2477, complete genome',\n",
       " 'NZ_CP015006.1 Aminobacter aminovorans strain KCTC 2477 plasmid pAA01, complete sequence',\n",
       " 'NZ_CP015007.1 Aminobacter aminovorans strain KCTC 2477 plasmid pAA02, complete sequence',\n",
       " 'NZ_CP015008.1 Aminobacter aminovorans strain KCTC 2477 plasmid pAA03, complete sequence',\n",
       " 'NZ_CP015009.1 Aminobacter aminovorans strain KCTC 2477 plasmid pAA04, complete sequence',\n",
       " 'NZ_CP010838.1 Bordetella pertussis strain H374, complete genome',\n",
       " 'NZ_CP010839.1 Bordetella pertussis strain H378, complete genome',\n",
       " 'NZ_CP010840.1 Bordetella pertussis strain H379, complete genome',\n",
       " 'NZ_CP014029.2 Klebsiella aerogenes strain FDAARGOS_152 chromosome, complete genome',\n",
       " 'NZ_CP014030.2 Citrobacter sp. FDAARGOS_156 strain FDAARGOS_155 chromosome, complete genome',\n",
       " 'NZ_CP014031.2 Hafnia paralvei strain FDAARGOS_158 chromosome, complete genome',\n",
       " 'NZ_CP014268.2 Escherichia coli B strain C2566, complete genome',\n",
       " 'NZ_CP014269.1 Escherichia coli B strain C3029, complete genome',\n",
       " 'NZ_CP014270.1 Escherichia coli K-12 strain DHB4, complete genome',\n",
       " 'NZ_CP014271.1 Escherichia coli K-12 strain DHB4 plasmid F128-(DHB4), complete sequence',\n",
       " 'NZ_CP014272.1 Escherichia coli K-12 strain C3026, complete genome',\n",
       " 'NZ_CP014273.1 Escherichia coli K-12 strain C3026 plasmid F128-(C3026), complete sequence',\n",
       " 'NZ_CP014326.1 Streptococcus mitis strain SVGS_061, complete genome',\n",
       " 'NZ_CP014223.1 [Clostridium] propionicum DSM 1682, complete genome',\n",
       " 'NZ_CP014279.1 Corynebacterium stationis strain ATCC 6872, complete genome',\n",
       " 'NZ_CP013926.1 Alteromonas stellipolaris strain LMG 21861, complete genome',\n",
       " 'NZ_CP013927.1 Alteromonas stellipolaris strain LMG 21861 plasmid pASTE61-200, complete sequence',\n",
       " 'NZ_CP013932.1 Alteromonas sp. Mac1, complete genome',\n",
       " 'NZ_CP014322.1 Alteromonas addita strain R10SW13, complete genome',\n",
       " 'NZ_CP014274.1 Stenotrophomonas sp. KCTC 12332, complete genome',\n",
       " 'NZ_CP014323.1 Alteromonas macleodii strain D7, complete genome',\n",
       " 'NZ_CP014301.1 Bosea sp. PAMC 26642, complete genome',\n",
       " 'NZ_CP014302.1 Bosea sp. PAMC 26642 plasmid, complete sequence',\n",
       " 'NZ_CP014304.1 Hymenobacter sp. PAMC26628, complete genome',\n",
       " 'NZ_CP014303.1 Hymenobacter sp. PAMC26628 plasmid, complete sequence',\n",
       " 'NZ_CP013928.1 Alteromonas mediterranea strain UM8, complete genome',\n",
       " 'NZ_CP013929.1 Alteromonas mediterranea strain UM8 plasmid pAMEDUM8_300, complete sequence',\n",
       " 'NZ_CP014206.1 Pseudodesulfovibrio indicus strain J2 chromosome, complete genome',\n",
       " 'NZ_CP005083.1 Sphingobium sp. TKS chromosome 1, complete sequence',\n",
       " 'NZ_CP005084.1 Sphingobium sp. TKS chromosome 2, complete sequence',\n",
       " 'NZ_CP005085.1 Sphingobium sp. TKS plasmid pTK1, complete sequence',\n",
       " 'NZ_CP005086.1 Sphingobium sp. TKS plasmid pTK2, complete sequence',\n",
       " 'NZ_CP005087.1 Sphingobium sp. TKS plasmid pTK3, complete sequence',\n",
       " 'NZ_CP005088.1 Sphingobium sp. TKS plasmid pTK4, complete sequence',\n",
       " 'NZ_CP005090.1 Sphingobium sp. TKS plasmid pTK6, complete sequence',\n",
       " 'NZ_CP005091.1 Sphingobium sp. TKS plasmid pTK7, complete sequence',\n",
       " 'NZ_CP005092.1 Sphingobium sp. TKS plasmid pTK8, complete sequence',\n",
       " 'NZ_CP005093.1 Sphingobium sp. TKS plasmid pTK9, complete sequence',\n",
       " 'NZ_CP005089.1 Sphingobium sp. TKS plasmid pTK5, complete sequence',\n",
       " 'NZ_CP005188.1 Sphingobium sp. MI1205 chromosome 1, complete sequence',\n",
       " 'NZ_CP005189.1 Sphingobium sp. MI1205 chromosome 2, complete sequence',\n",
       " 'NZ_CP005190.1 Sphingobium sp. MI1205 plasmid pMI1, complete sequence',\n",
       " 'NZ_CP005191.1 Sphingobium sp. MI1205 plasmid pMI2, complete sequence',\n",
       " 'NZ_CP005192.1 Sphingobium sp. MI1205 plasmid pMI3, complete sequence',\n",
       " 'NZ_CP005193.1 Sphingobium sp. MI1205 plasmid pMI4, complete sequence',\n",
       " 'NZ_CP014004.1 Klebsiella pneumoniae subsp. pneumoniae strain NUHL24835, complete genome',\n",
       " 'NZ_CP014005.1 Klebsiella pneumoniae subsp. pneumoniae strain NUHL24835 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP014006.1 Klebsiella pneumoniae subsp. pneumoniae strain NUHL24835 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP013357.1 Burkholderia oklahomensis EO147 chromosome 1, complete sequence',\n",
       " 'NZ_CP013356.1 Burkholderia oklahomensis EO147 chromosome 2, complete sequence',\n",
       " 'NZ_CP013358.1 Burkholderia oklahomensis C6786 chromosome 1, complete sequence',\n",
       " 'NZ_CP013359.1 Burkholderia oklahomensis C6786 chromosome 2, complete sequence',\n",
       " 'NZ_CP013373.1 Burkholderia sp. NRF60-BP8 chromosome 1, complete sequence',\n",
       " 'NZ_CP013372.1 Burkholderia sp. NRF60-BP8 chromosome 2, complete sequence',\n",
       " 'NZ_CP013374.1 Burkholderia sp. NRF60-BP8 chromosome 3, complete sequence',\n",
       " 'NZ_CP013389.1 Burkholderia sp. BDU8 chromosome 1, complete sequence',\n",
       " 'NZ_CP013388.1 Burkholderia sp. BDU8 chromosome 2, complete sequence',\n",
       " 'NZ_CP013386.1 Burkholderia sp. BDU6 chromosome 1, complete sequence',\n",
       " 'NZ_CP013387.1 Burkholderia sp. BDU6 chromosome 2, complete sequence',\n",
       " 'NZ_CP013384.1 Burkholderia sp. LA-2-3-30-S1-D2 chromosome 1, complete sequence',\n",
       " 'NZ_CP013383.1 Burkholderia sp. LA-2-3-30-S1-D2 chromosome 2, complete sequence',\n",
       " 'NZ_CP013385.1 Burkholderia sp. LA-2-3-30-S1-D2 chromosome 3, complete sequence',\n",
       " 'NZ_CP013417.1 Burkholderia sp. MSMB0266 chromosome 1, complete sequence',\n",
       " 'NZ_CP013418.1 Burkholderia sp. MSMB0266 chromosome 2, complete sequence',\n",
       " 'NZ_CP013419.1 Burkholderia sp. MSMB0266 plasmid pMSMB0266, complete sequence',\n",
       " 'NZ_CP013424.1 Burkholderia sp. MSMB0852 chromosome 1, complete sequence',\n",
       " 'NZ_CP013425.1 Burkholderia sp. MSMB0852 chromosome 2, complete sequence',\n",
       " 'NZ_CP013423.1 Burkholderia sp. MSMB0852 plasmid pMSMB0852, complete sequence',\n",
       " 'NZ_CP013427.1 Burkholderia sp. MSMB0856 chromosome 1, complete sequence',\n",
       " 'NZ_CP013428.1 Burkholderia sp. MSMB0856 chromosome 2, complete sequence',\n",
       " 'NZ_CP013429.1 Burkholderia sp. MSMB0856 chromosome 3, complete sequence',\n",
       " 'NZ_CP013426.1 Burkholderia sp. MSMB0856 plasmid pMSMB0856, complete sequence',\n",
       " 'NZ_CP014010.1 Klebsiella pneumoniae subsp. pneumoniae strain RJF999, complete genome',\n",
       " 'NZ_CP014011.1 Klebsiella pneumoniae subsp. pneumoniae strain RJF999 plasmid pRJF999, complete sequence',\n",
       " 'NZ_CP014008.1 Klebsiella pneumoniae subsp. pneumoniae strain RJF293, complete genome',\n",
       " 'NZ_CP014009.1 Klebsiella pneumoniae subsp. pneumoniae strain RJF293 plasmid pRJF293, complete sequence',\n",
       " 'NZ_CP013457.1 Burkholderia sp. MSMB617WGS chromosome 1, complete sequence',\n",
       " 'NZ_CP013458.1 Burkholderia sp. MSMB617WGS chromosome 2, complete sequence',\n",
       " 'NZ_CP013992.1 Flavobacterium columnare strain 94-081, complete genome',\n",
       " 'NZ_CP013920.1 Arsenophonus symbiont of Lipoptena fortisetosa strain CB, complete genome',\n",
       " 'NZ_CP013975.1 Piscirickettsia salmonis strain CGR02, complete genome',\n",
       " 'NZ_CP013976.1 Piscirickettsia salmonis strain CGR02 plasmid pPSCRG02-1, complete sequence',\n",
       " 'NZ_CP013977.1 Piscirickettsia salmonis strain CGR02 plasmid pPSCRG02-2, complete sequence',\n",
       " 'NZ_CP013978.1 Piscirickettsia salmonis strain CGR02 plasmid pPSCRG02-4, complete sequence',\n",
       " 'NZ_CP013997.1 Pseudomonas monteilii strain USDA-ARS-USMARC-56711, complete genome',\n",
       " 'NC_022998.1 Spiroplasma apis B31, complete genome',\n",
       " 'NC_023037.2 Paenibacillus polymyxa CR1, complete genome',\n",
       " 'NZ_CP006579.1 Aeromonas hydrophila 4AK4, complete genome',\n",
       " 'NZ_CP006859.1 Listeria monocytogenes serotype 1/2a str. 08-6997, complete genome',\n",
       " 'NZ_CP003915.1 Advenella mimigardefordensis DPN7, complete genome',\n",
       " 'NZ_CP003916.1 Advenella mimigardefordensis DPN7 plasmid 24p, complete sequence',\n",
       " 'NZ_CP007128.1 Gemmatirosa kalamazoonesis strain KBS708, complete genome',\n",
       " 'NZ_CP007129.1 Gemmatirosa kalamazoonesis strain KBS708 plasmid 1, complete sequence',\n",
       " 'NZ_CP007127.1 Gemmatirosa kalamazoonesis strain KBS708 phage-like extrachromosal element, complete sequence',\n",
       " 'NZ_CP007130.1 Gemmatirosa kalamazoonesis strain KBS708 plasmid 2, complete sequence',\n",
       " 'NZ_CP006720.1 Spiroplasma mirum ATCC 29335 strain SMCA, complete genome',\n",
       " 'NZ_CP006715.1 Bifidobacterium breve 689b, complete genome',\n",
       " 'NZ_CP007230.1 Yersinia similis strain 228, complete genome',\n",
       " 'NZ_CP007231.1 Yersinia similis strain 228 plasmid, complete sequence',\n",
       " 'NC_022107.1 Chlamydia trachomatis F/11-96, complete genome',\n",
       " 'NZ_CP007446.1 Snodgrassella alvi wkB2, complete genome',\n",
       " 'NZ_CP012371.1 Nitrosospira briensis C-128, complete genome',\n",
       " 'NZ_CP010493.1 Campylobacter jejuni strain CJ677CC062, complete genome',\n",
       " 'NZ_CP010494.1 Campylobacter jejuni strain CJ677CC059, complete genome',\n",
       " 'NZ_CP010496.1 Campylobacter jejuni strain CJ677CC032, complete genome',\n",
       " 'NZ_CP010497.1 Campylobacter jejuni strain CJ677CC033, complete genome',\n",
       " 'NZ_CP010498.1 Campylobacter jejuni strain CJ677CC537, complete genome',\n",
       " 'NZ_CP010499.1 Campylobacter jejuni strain CJ677CC542, complete genome',\n",
       " 'NZ_CP010500.1 Campylobacter jejuni strain CJ677CC528, complete genome',\n",
       " 'NZ_CP010495.1 Campylobacter jejuni strain CJ677CC538, complete genome',\n",
       " 'NZ_CP010501.1 Campylobacter jejuni strain CJ677CC520, complete genome',\n",
       " 'NZ_CP010502.1 Campylobacter jejuni strain CJ677CC014, complete genome',\n",
       " 'NZ_CP010503.1 Campylobacter jejuni strain CJ677CC039, complete genome',\n",
       " 'NZ_CP010504.1 Campylobacter jejuni strain CJ677CC085, complete genome',\n",
       " 'NZ_CP010505.1 Campylobacter jejuni strain CJ677CC052, complete genome',\n",
       " 'NZ_CP010506.1 Campylobacter jejuni strain CJ677CC527, complete genome',\n",
       " 'NZ_CP010507.1 Campylobacter jejuni strain CJ677CC078, complete genome',\n",
       " 'NZ_CP010508.1 Campylobacter jejuni strain CJ677CC523, complete genome',\n",
       " 'NZ_CP010509.1 Campylobacter jejuni strain CJ677CC540, complete genome',\n",
       " 'NZ_CP010510.1 Campylobacter jejuni strain CJ677CC040, complete genome',\n",
       " 'NZ_CP010511.1 Campylobacter jejuni strain CJ677CC061, complete genome',\n",
       " 'NZ_CP010457.1 Campylobacter jejuni strain CJ677CC539, complete genome',\n",
       " 'NZ_CP010458.1 Campylobacter jejuni strain CJ677CC533, complete genome',\n",
       " 'NZ_CP010459.1 Campylobacter jejuni strain CJ677CC047, complete genome',\n",
       " 'NZ_CP009802.1 Streptomyces sp. FR-008, complete genome',\n",
       " 'NZ_CP009803.1 Streptomyces sp. FR-008 plasmid pSSFR1, complete sequence',\n",
       " 'NZ_CP009804.1 Streptomyces sp. FR-008 plasmid pSSFR2, complete sequence',\n",
       " 'NZ_CP012482.1 Bacillus cellulasensis strain NJ-V2, complete genome',\n",
       " 'NZ_CP013114.1 Staphylococcus equorum strain KS1039, complete genome',\n",
       " 'NZ_CP009847.1 Vibrio parahaemolyticus strain FORC_004 chromosome 1, complete sequence',\n",
       " 'NZ_CP009848.1 Vibrio parahaemolyticus strain FORC_004 chromosome 2, complete sequence',\n",
       " 'NZ_CP009849.1 Vibrio parahaemolyticus strain FORC_004 plasmid pFORC4, complete sequence',\n",
       " 'NZ_CP009984.1 Vibrio vulnificus strain FORC_009 chromosome 1, complete sequence',\n",
       " 'NZ_CP009985.1 Vibrio vulnificus strain FORC_009 chromosome 2, complete sequence',\n",
       " 'NZ_CP012837.1 Corynebacterium pseudotuberculosis strain 1002B, complete genome',\n",
       " 'NZ_CP012907.1 Helicobacter pylori strain 29CaP, complete genome',\n",
       " 'NZ_CP012905.1 Helicobacter pylori strain 7C, complete sequence',\n",
       " 'NZ_CP012906.1 Helicobacter pylori strain 7C plasmid, complete sequence',\n",
       " 'NZ_CP013120.1 Alteromonas stellipolaris LMG 21856, complete genome',\n",
       " 'NZ_CP013121.1 Fusobacterium nucleatum subsp. polymorphum strain ChDC F306, complete genome',\n",
       " 'NZ_CP013122.1 Fusobacterium nucleatum subsp. polymorphum strain ChDC F306 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP013123.1 Fusobacterium nucleatum subsp. polymorphum strain ChDC F306 plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP014140.1 Aneurinibacillus sp. XH2, complete genome',\n",
       " 'NZ_CP013126.1 Acidipropionibacterium acidipropionici strain CGMCC 1.2230 chromosome, complete genome',\n",
       " 'NZ_CP012921.1 Salmonella enterica subsp. enterica serovar Heidelberg strain SA02DT10168701, complete genome',\n",
       " 'NZ_CP012922.1 Salmonella enterica subsp. enterica serovar Heidelberg strain SA02DT10168701 plasmid pSA02DT10168701_37, complete sequence',\n",
       " 'NZ_CP012923.1 Salmonella enterica subsp. enterica serovar Heidelberg strain SA02DT10168701 plasmid pSA02DT10168701_99, complete sequence',\n",
       " 'NZ_CP012924.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374, complete genome',\n",
       " 'NZ_CP012925.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374 plasmid p12-4374_2, complete sequence',\n",
       " 'NZ_CP012927.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374 plasmid p12-4374_3, complete sequence',\n",
       " 'NZ_CP012926.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374 plasmid p12-4374_37, complete sequence',\n",
       " 'NZ_CP012928.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374 plasmid p12-4374_62, complete sequence',\n",
       " 'NZ_CP012929.1 Salmonella enterica subsp. enterica serovar Heidelberg strain 12-4374 plasmid p12-4374_96, complete sequence',\n",
       " 'NZ_CP012930.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290, complete genome',\n",
       " 'NZ_CP012932.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_2, complete sequence',\n",
       " 'NZ_CP012931.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_23, complete sequence',\n",
       " 'NZ_CP012933.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_3-1, complete sequence',\n",
       " 'NZ_CP012934.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_3-2, complete sequence',\n",
       " 'NZ_CP012935.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_3-3, complete sequence',\n",
       " 'NZ_CP012936.1 Salmonella enterica subsp. enterica serovar Heidelberg strain N13-01290 plasmid pN13-01290_98, complete sequence',\n",
       " 'NZ_CP013133.1 Rickettsia rhipicephali strain HJ#5, complete genome',\n",
       " 'NZ_CP013134.1 Rickettsia rhipicephali strain HJ#5 plasmid pHJ51, complete sequence',\n",
       " 'NZ_CP013135.1 Rickettsia rhipicephali strain HJ#5 plasmid pHJ52 sequence',\n",
       " 'NZ_CP013112.1 Escherichia coli strain YD786, complete genome',\n",
       " 'NZ_CP013141.1 Lysobacter antibioticus strain ATCC 29479 genome',\n",
       " 'NZ_CP011129.1 Lysobacter antibioticus strain 76, complete genome',\n",
       " 'NZ_CP012096.1 Serratia plymuthica strain 3Rp8, complete genome',\n",
       " 'NZ_CP012097.1 Serratia plymuthica strain 3Re4-18, complete genome',\n",
       " 'NZ_CP015963.1 Altererythrobacter ishigakiensis strain NBRC 107699 chromosome, complete genome',\n",
       " 'NZ_CP016033.1 Porphyrobacter neustonensis strain DSM 9434, complete genome',\n",
       " 'NZ_CP015990.1 Klebsiella pneumoniae strain BR chromosome, complete genome',\n",
       " 'NZ_CP015991.1 Klebsiella pneumoniae strain BR plasmid pWSZBR, complete sequence',\n",
       " 'NZ_CP014762.1 Klebsiella pneumoniae strain KPNIH39, complete genome',\n",
       " 'NZ_CP014763.1 Klebsiella pneumoniae strain KPNIH39 plasmid pKPN-332, complete sequence',\n",
       " 'NZ_CP014764.1 Klebsiella pneumoniae strain KPNIH39 plasmid pKPN-704, complete sequence',\n",
       " 'NZ_CP014765.1 Klebsiella pneumoniae strain KPNIH39 plasmid pKpQIL-9b8, complete sequence',\n",
       " 'NZ_CP012968.1 Staphylococcus sp. AntiMn-1, complete genome',\n",
       " 'NZ_CP013300.1 Staphylococcus sp. AntiMn-1 plasmid AM1_232, complete sequence',\n",
       " 'NZ_CP015850.1 Ralstonia solanacearum strain YC40-M chromosome, complete genome',\n",
       " 'NZ_CP015851.1 Ralstonia solanacearum strain YC40-M plasmid, complete sequence',\n",
       " 'NZ_CP015025.1 Klebsiella pneumoniae strain Kpn223, complete genome',\n",
       " 'NZ_CP015026.1 Klebsiella pneumoniae strain Kpn223 plasmid pKPN-065, complete sequence',\n",
       " 'NZ_CP015130.1 Klebsiella pneumoniae strain Kpn555 chromosome, complete genome',\n",
       " 'NZ_CP015131.1 Klebsiella pneumoniae strain Kpn555 plasmid pKPN-7c3, complete sequence',\n",
       " 'NZ_CP015133.1 Klebsiella pneumoniae strain Kpn555 plasmid pKPN-d6b, complete sequence',\n",
       " 'NZ_CP015132.1 Klebsiella pneumoniae strain Kpn555 plasmid pKPN-d90, complete sequence',\n",
       " 'NZ_CP015159.1 Escherichia coli strain Eco889 chromosome, complete genome',\n",
       " 'NZ_CP015161.1 Escherichia coli strain Eco889 plasmid pECO-93a, complete sequence',\n",
       " 'NZ_CP015160.1 Escherichia coli strain Eco889 plasmid pECO-fce, complete sequence',\n",
       " 'NZ_CP015309.1 Corynebacterium pseudotuberculosis strain PA02 chromosome, complete genome',\n",
       " 'NZ_CP014623.1 Lactobacillus backii strain TMW 1.1988 chromosome, complete genome',\n",
       " 'NZ_CP014624.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-1, complete sequence',\n",
       " 'NZ_CP014625.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-10, complete sequence',\n",
       " 'NZ_CP014626.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-2, complete sequence',\n",
       " 'NZ_CP014627.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-3, complete sequence',\n",
       " 'NZ_CP014628.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-4, complete sequence',\n",
       " 'NZ_CP014629.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-5, complete sequence',\n",
       " 'NZ_CP014630.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-6, complete sequence',\n",
       " 'NZ_CP014631.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-7, complete sequence',\n",
       " 'NZ_CP014632.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-8, complete sequence',\n",
       " 'NZ_CP014633.1 Lactobacillus backii strain TMW 1.1988 plasmid L11988-9, complete sequence',\n",
       " 'NZ_CP012969.1 Psychrobacter sp. AntiMn-1, complete genome',\n",
       " 'NZ_CP013101.1 Clostridium perfringens strain FORC_025, complete genome',\n",
       " 'NZ_CP017715.1 Marinobacter salinus strain Hb8 chromosome, complete genome',\n",
       " 'NZ_CP017188.2 Xanthomonas citri pv. glycines str. 8ra chromosome, complete genome',\n",
       " 'NZ_CP017189.2 Xanthomonas citri pv. glycines str. 8ra plasmid pXAG27.8ra, complete sequence',\n",
       " 'NZ_CP017190.1 Xanthomonas campestris pv. vesicatoria str. 85-10, complete sequence',\n",
       " 'NZ_CP017191.1 Xanthomonas campestris pv. vesicatoria str. 85-10 plasmid p_XCV_1, complete sequence',\n",
       " 'NZ_CP017192.1 Xanthomonas campestris pv. vesicatoria str. 85-10 plasmid p_XCV_2, complete sequence',\n",
       " 'NZ_CP017193.1 Xanthomonas campestris pv. vesicatoria str. 85-10 plasmid p_XCV_3, complete sequence',\n",
       " 'NZ_CP017603.1 Clostridium formicaceticum strain ATCC 27076, complete genome',\n",
       " 'NZ_CP015208.1 Candidatus Rhodoluna planktonica strain MWH-Dar1, complete genome',\n",
       " 'NZ_CP015381.1 Desulfococcus multivorans strain DSM 2059, complete genome',\n",
       " 'NZ_CP017751.1 Cupriavidus sp. USMAHM13 chromosome 1, complete sequence',\n",
       " 'NZ_CP017752.1 Cupriavidus sp. USMAHM13 chromosome 2, complete sequence',\n",
       " 'NZ_CP017753.1 Cupriavidus sp. USMAHM13 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP017748.1 Cupriavidus sp. USMAA2-4 chromosome 1, complete sequence',\n",
       " 'NZ_CP017749.1 Cupriavidus sp. USMAA2-4 chromosome 2, complete sequence',\n",
       " 'NZ_CP017750.1 Cupriavidus sp. USMAA2-4 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP017754.1 Cupriavidus sp. USMAA1020 chromosome 1, complete sequence',\n",
       " 'NZ_CP017755.1 Cupriavidus sp. USMAA1020 chromosome 2, complete sequence',\n",
       " 'NZ_CP017756.1 Cupriavidus sp. USMAA1020 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP017747.1 Bacillus velezensis strain SYBC H47 chromosome, complete genome',\n",
       " 'NZ_CP017421.1 Arthrobacter sp. ZXY-2 chromosome, complete genome',\n",
       " 'NZ_CP017422.1 Arthrobacter sp. ZXY-2 plasmid pZXY21, complete sequence',\n",
       " 'NZ_CP017423.1 Arthrobacter sp. ZXY-2 plasmid pZXY22, complete sequence',\n",
       " 'NZ_CP017424.1 Arthrobacter sp. ZXY-2 plasmid pZXY23, complete sequence',\n",
       " 'NZ_CP017425.1 Arthrobacter sp. ZXY-2 plasmid pZXY24, complete sequence',\n",
       " 'NZ_CP017426.1 Arthrobacter sp. ZXY-2 plasmid pZXY25, complete sequence',\n",
       " 'NZ_CP017640.1 Methylobacterium sp. C1 chromosome, complete genome',\n",
       " 'NZ_CP017718.1 Hyphomonas sp. Mor2 genome',\n",
       " 'NZ_CP013013.1 Vibrio cholerae strain Env-390 chromosome 1, complete sequence',\n",
       " 'NZ_CP013014.1 Vibrio cholerae strain Env-390 chromosome 2, complete sequence',\n",
       " 'NZ_CP013475.1 Mycobacterium tuberculosis strain 1458, complete genome',\n",
       " 'NZ_CP017707.1 Chromobacterium vaccinii strain 21-1 chromosome, complete genome',\n",
       " 'NZ_CP017622.1 Ketogulonicigenium vulgare strain SPU B805 chromosome, complete genome',\n",
       " 'NZ_CP012426.1 Klebsiella pneumoniae strain KP5 chromosome, complete genome',\n",
       " 'NZ_CP012427.1 Klebsiella pneumoniae strain KP5 plasmid pSg1-1, complete sequence',\n",
       " 'NZ_CP012428.1 Klebsiella pneumoniae strain KP5 plasmid pSg1-2, complete sequence',\n",
       " 'NZ_CP012429.1 Klebsiella pneumoniae strain KP5 plasmid pSg1-3, complete sequence',\n",
       " 'NZ_CP011839.1 Klebsiella pneumoniae strain KP5 plasmid pSg1-NDM, complete sequence',\n",
       " 'NZ_CP017100.1 Escherichia coli strain K-12 NEB 5-alpha chromosome, complete genome',\n",
       " 'NZ_CP017107.1 Lactobacillus salivarius strain CICC 23174 chromosome, complete genome',\n",
       " 'NZ_CP017108.1 Lactobacillus salivarius strain CICC23174 plasmid pLS_1 sequence',\n",
       " 'NZ_CP017109.1 Lactobacillus salivarius strain CICC23174 plasmid pLS_2 sequence',\n",
       " 'NZ_CP017110.1 Lactobacillus salivarius strain CICC23174 plasmid pLS_3, complete sequence',\n",
       " 'NZ_CP017124.1 Lactobacillus curvatus strain WiKim38, complete genome',\n",
       " 'NZ_CP016395.1 Bacillus velezensis strain M75 chromosome, complete genome',\n",
       " 'NZ_CP017111.1 Sulfurospirillum halorespirans DSM 13726 chromosome, complete genome',\n",
       " 'NZ_CP017116.1 Erysipelothrix rhusiopathiae strain WH13013 chromosome, complete genome',\n",
       " 'NZ_CP012692.1 Staphylococcus aureus strain FORC_027, complete genome',\n",
       " 'NZ_CP016019.1 Bifidobacterium longum subsp. longum strain AH1206, complete genome',\n",
       " 'NZ_CP017247.1 Bacillus licheniformis strain BL1202, complete genome',\n",
       " 'NZ_CP016385.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931, complete genome',\n",
       " 'NZ_CP016386.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931 plasmid p931-3904, complete sequence',\n",
       " 'NZ_CP016387.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931 plasmid p931IncI1, complete sequence',\n",
       " 'NZ_CP016388.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931 plasmid p931IncI2, complete sequence',\n",
       " 'NZ_CP016389.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931 plasmid pESBL931, complete sequence',\n",
       " 'NZ_CP016390.1 Salmonella enterica subsp. enterica serovar Typhimurium strain 13-931 plasmid pSLT931, complete sequence',\n",
       " 'NZ_CP015217.1 Leptospira alstonii strain GWTS #1 chromosome 1, complete sequence',\n",
       " 'NZ_CP015218.1 Leptospira alstonii strain GWTS #1 chromosome 2, complete sequence',\n",
       " 'NZ_CP016044.1 Edwardsiella piscicida strain S11-285 chromosome, complete genome',\n",
       " 'NZ_CP016445.1 Edwardsiella piscicida strain S11-285 plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP017157.1 Streptomyces lydicus strain 103 chromosome, complete genome',\n",
       " 'NZ_CP017149.1 Pseudomonas aeruginosa strain ATCC 15692 chromosome, complete genome',\n",
       " 'NZ_CP017150.1 Brevibacterium aurantiacum strain SMQ-1335 chromosome, complete genome',\n",
       " 'NZ_CP017175.1 Burkholderia mallei strain Bahrain1 chromosome 1, complete sequence',\n",
       " 'NZ_CP017176.1 Burkholderia mallei strain Bahrain1 chromosome 2, complete sequence',\n",
       " 'NZ_CP016894.1 Bacillus subtilis strain HJ0-6, complete genome',\n",
       " 'NZ_CP016895.1 Acinetobacter larvae strain BRTC-1 chromosome, complete genome',\n",
       " 'NZ_CP016294.1 Stenotrophomonas rhizophila strain QL-P4, complete genome',\n",
       " 'NZ_CP016824.1 Streptomyces sampsonii strain KJ40 chromosome, complete genome',\n",
       " 'NZ_CP016341.1 Bordetella holmesii strain H903, complete genome',\n",
       " 'NZ_CP016811.1 Klebsiella pneumoniae strain DHQP1002001 chromosome, complete genome',\n",
       " 'NZ_CP016810.1 Klebsiella pneumoniae strain DHQP1002001 plasmid p_IncFIB_DHQP1002001, complete sequence',\n",
       " 'NZ_CP016812.1 Klebsiella pneumoniae strain DHQP1002001 plasmid p_incR_DHQP1002001, complete sequence',\n",
       " 'NZ_CP016342.1 Bordetella parapertussis strain H904, complete genome',\n",
       " 'NZ_CP011533.1 Streptomyces noursei ATCC 11455, complete genome',\n",
       " 'NZ_CP012334.1 Bordetella sp. H567, complete genome',\n",
       " 'NZ_CP013749.1 Lactobacillus plantarum strain KP, complete genome',\n",
       " 'NZ_CP013750.1 Lactobacillus plantarum strain KP plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP013751.1 Lactobacillus plantarum strain KP plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP013752.1 Lactobacillus plantarum strain KP plasmid unnamed3, complete sequence',\n",
       " 'NZ_CP013753.1 Lactobacillus plantarum strain DF, complete genome',\n",
       " 'NZ_CP013754.1 Lactobacillus plantarum strain DF plasmid unnamed1, complete sequence',\n",
       " 'NZ_CP013755.1 Lactobacillus plantarum strain DF plasmid unnamed2, complete sequence',\n",
       " 'NZ_CP013756.1 Lactobacillus plantarum strain DF plasmid unnamed3, complete sequence',\n",
       " 'NZ_CP016431.1 Bordetella bronchiseptica strain I328 chromosome, complete genome',\n",
       " 'NZ_CP029397.1 Acinetobacter defluvii strain WCHA30 chromosome, complete genome',\n",
       " 'NZ_CP029389.1 Acinetobacter defluvii strain WCHA30 plasmid p1_010030, complete sequence',\n",
       " 'NZ_CP029390.1 Acinetobacter defluvii strain WCHA30 plasmid p2_010030, complete sequence',\n",
       " 'NZ_CP029391.1 Acinetobacter defluvii strain WCHA30 plasmid p3_010030, complete sequence',\n",
       " 'NZ_CP029392.1 Acinetobacter defluvii strain WCHA30 plasmid p4_010030, complete sequence',\n",
       " 'NZ_CP029393.1 Acinetobacter defluvii strain WCHA30 plasmid p5_010030, complete sequence',\n",
       " 'NZ_CP029394.1 Acinetobacter defluvii strain WCHA30 plasmid p6_010030, complete sequence',\n",
       " 'NZ_CP029395.1 Acinetobacter defluvii strain WCHA30 plasmid p7_010030, complete sequence',\n",
       " 'NZ_CP029396.1 Acinetobacter defluvii strain WCHA30 plasmid pOXA58_010030, complete sequence',\n",
       " 'NZ_CP012375.1 Microcystis aeruginosa NIES-2481, complete genome',\n",
       " 'NZ_CP025929.1 Microcystis aeruginosa NIES-2481 plasmid p1, complete sequence',\n",
       " ...]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_fastas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n_sequences</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>genus</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Bacillus</th>\n",
       "      <td>1132</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Streptomyces</th>\n",
       "      <td>743</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vibrio</th>\n",
       "      <td>468</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rhizobium</th>\n",
       "      <td>325</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudomonas</th>\n",
       "      <td>304</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus</th>\n",
       "      <td>301</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Clostridium</th>\n",
       "      <td>259</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Streptococcus</th>\n",
       "      <td>222</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Planktothrix</th>\n",
       "      <td>179</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stenotrophomonas</th>\n",
       "      <td>176</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Escherichia</th>\n",
       "      <td>133</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Paenibacillus</th>\n",
       "      <td>127</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Listeria</th>\n",
       "      <td>104</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corynebacterium</th>\n",
       "      <td>103</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Klebsiella</th>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Shigella</th>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salmonella</th>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Enterobacter</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  n_sequences  species\n",
       "genus                                 \n",
       "Bacillus                 1132       11\n",
       "Streptomyces              743        5\n",
       "Vibrio                    468        5\n",
       "Rhizobium                 325        6\n",
       "Pseudomonas               304        8\n",
       "Staphylococcus            301        6\n",
       "Clostridium               259        5\n",
       "Streptococcus             222        6\n",
       "Planktothrix              179        5\n",
       "Stenotrophomonas          176        5\n",
       "Escherichia               133        3\n",
       "Paenibacillus             127        3\n",
       "Listeria                  104        4\n",
       "Corynebacterium           103        7\n",
       "Klebsiella                 16        3\n",
       "Shigella                   14        3\n",
       "Salmonella                  4        2\n",
       "Enterobacter                1        1"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts = inventory.reset_index().groupby(\"genus\").agg({\"count\", sum}).drop((\"species\"), axis=1)\n",
    "counts.columns=[\"n_sequences\",\"species\"]\n",
    "counts.sort_values(\"n_sequences\", ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
